---
title: 'Points of Confusion: Understanding "Typologically
  Awkward" Assemblages in the Final Palaeolithic'
author:
- Christian Steven Hoggard (University of Southampton, UK)
- Felix Riede (Aarhus University, DK)
date: "`r format(Sys.time(), '%B %Y')`"
output:
  word_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r packages, echo = FALSE, message = FALSE, warning = FALSE}
library(rio)
library(tidyverse)
library(gt)
library(doBy)
library(FactoMineR)
library(factoextra)
library(diverse)
library(vegan)
library(tidymodels)
library(broom)
library(tidytext)
library(rnaturalearth)
library(rnaturalearthdata)
library(ggrepel)
library(ggspatial)
library(extrafont)
```

# Introduction (Text needs updating)

The Late Glacial, defined as occurring from the first pronounced warming following the Last Glacial Maximum (LGM) to the beginning of the Holocene (c. 16,000-11,700 cal BP), is traditionally viewed as a period of great cultural diversity in northern Europe, with groups belonging to the Magdalenian and Epigravettian beginning to migrate northwards (Gamble et al. 2005; Riede, 2014; Wygal and Heidenreich, 2014). During this phase, groups occupying particular regions in northern Europe began to develop consistent differences in their toolkits, with variances in shouldered points (e.g. the Hamburgian: Burdukiewicz, 1986), backed points (e.g. the Federmessergruppen and Azilian: Schild, 1996), large tanged points (e.g. the Bromme: Taute, 1968) and small tanged points (e.g. the Ahrensburgian and Swiderian: Taute, 1968; Kobusiewicz, 2002).
As highlighted by Ivanovaite et al. (2019), these regional cultural entities provide the framework for understanding and inferring patterns and processes of migration, adaptation, and potentially even emerging ethnicity for this period. Cultural units are not static, and once their attendant territories are defined, cultures can move around, their economic focus can change, and cultures can interact. It is for this reason that the identification of robust archaeological taxonomic units are a fundamental precondition in our archaeological understanding of this period; without these units, the goal of understanding such past processes cannot be realised (Roberts and Vander Linden, 2011).  

Yet, techno-typological classifications for the European Palaeolithic (and further afield) have been repeatedly critiqued (Otte and Keeley, 1990; Neeley and Barton, 1994; Barton and Neeley, 1996; Felgenhauer, 1996; Clark, 1999, Vasil'ev, 2001; 2009; Tomášková, 2013; Clark and Riel-Salvatore, 2006; Shea, 2014; Lisitsyn, 2017; Ivanovaitė and Riede, 2018; Riede et al. 2019; Reynolds and Riede, 2019; Ivanovaite et al. 2019). Otte and Keeley (1990) note that many of the taxonomic units for the Final Palaeolithic, the *fossils directeurs* for ethno-geographic variability, are usually rooted in early antiquarian excavations, consisting of only a few key sites (as reflected in the practice of naming archaeological units after *loci classici*), while Sauer and Riede (2019) have highlighted that, for the Final Palaeolithic of Central Europe, units appear to reflect differing research histories more than empirical variation between assemblages. A more acerbic critique has been voiced by Houtsma et al. 1996: 143), arguing that '[o]nly when researchers of the Late Palaeolithic habitation of the Northwest European Plain escape the constraints of contemporary national borders and the paradigmatic straight-jackets of provincialism and regional chauvinism, which lead to insularity, will we be in a position to gain analytical control of the totality of extant data partitioned into uniform and mutually comparable sets of demonstrably relevant attributes' (Houtsma et al. 1996: 143). With a lack of genetic material available to contribute to the understanding of the chronologies, movements and behaviours of Final Palaeolithic communities, stone tool variants remain the premier proxies for inferring population structure.  

Attempts are currently underway to provide a more objective framework for the Final Palaeolithic, founded on conceptualising material culture as an underwritten system of information transmission across generations, through the production of reproducible, objective phylogenetic assessments of archaeological taxonomic units (Riede et al. 2019; Ivanovaite et al. 2019). This article, through the same theoretical framework, considers this problem on a micro-level, specifically that of ‘typologically awkward’ assemblages, i.e. archaeological contexts with multiple techno-typological identifiers and reflecting one cultural units. In considering the Final Palaeolithic contexts of Häcklingen and Bienenbuttel (Germany), this article will highlight how an empirical technological and phylogenetic assessment can shed light on their positioning within the wider landscape and context, and further highlight some of the issues in techno-typological classifications.  


# Site Overviews: Häcklingen and Bienenbuttel

## Bienenbüttel FStNr. 15 (Lower Saxony, Germany)

Bienenbüttel-15 is situated on the north-east bank of the Ilmenau river, between the villages of Bienenbüttel and Wichmannsburg, and is approximately 11km to the south-east of Häcklingen (discussed below). Following test cuttings in autumn 1997, a more detailed investigation commenced in 1998 by the Bezirksarchiiologie Lüneburg Ausgrabungen (Lüneburg Distict Archaeological Unit), following proposed building development in the area.  

Excavations revealed a wealth of Neolithic material including post features, ceramic material in addition to a cooking pit; later prehistoric material from the Bronze Age was also recovered (Richter 2001). Below, a number of backed points were identified, in addition a number of scrapers, blade cores, burins and blades. A number of  fragmented and complete Federmesser and Creswellian points (Figure 1) were identified in addition to three Zonhoven points (sensu Schwabedissen 1954), blades typical of the Ahrensburgian featuring sloping retouched tips. One Hamburgian point, belonging to the Havelte Group (sensu Bohmers 1956 and Stapert 1997) was also identified among the assemblage. A clear temporal separation between the Final Palaeolithic and Neolithic material was acknowledged, with little mechanical redeposition assumed, however an absolute chronometric framework was not generated.  

Richter (2001) first supposed, through the identification of retouched tips, and the end-scraper component of the assemblage that elements from the both Wehlen and Tjonger Federmesser sub-groups (*sensu* Schwabedissen 1954) are identified. However, with the clear presence of Creswellian (and Ahrensburg and Hamburgian elements), Richter (2001) queried the pre-existing framework, following similar doubts by Paddaya (1971) and Ikinger (1998), and stressed the lack of precise qualitative and quantitative analyses in understanding group structure (Richter, 2001: 34).  

For further details on the site of Bienenbüttel FStNr. 15 please refer to Richter (2001).    

## Häcklingen FStNr. 19 (Lower Saxony, Germany)  

Excavations close to the village of Häcklingen, situated near the town of Lüneburg (Lower Saxony, Germany), first began from 1983-1985, following the identification of Palaeolithic and Neolithic material during routine surface inspection (Assendorp, 1997). Following the identification of Neolithic pit features, ceramic material, burned bone fragments and hazelnut shells, student excavations led by the University of Hamburg and University of Prague were undertaken towards the end of the twentieth century. 

With ice wedge formations thought to have occurred at this site, as described elsewhere for northern Germany (Clausen, 1995, 1997; Zoller, 1981), and artefacts separated from discoloured areas in the soil, a vertical shift in archaeological material is hypothesised (Richter, 2001). This is evident in the examination of the Final Palaeolithic material, with patination and minor edge damage contrasting the fresh-edged and unrelocated Neolithic material, and contrasts the sequence at Bienenbüttel-15.  

Among the Final Palaeolithic material were a variety of  tanged point and backed point forms, including Ahrensburgian, Lyngby and Chwalibogowice tanged points, in addition to a number of Penknife backed points (see Figure 1). Richter (2002) notes that classically (*sensu* Schwabedissen 1954) Häcklingen can be attributed to the Rissener sub-group of the Federmessergruppen, before highlighting that the Rissener categorisation, and the sub-categories of the Federmesser more generally, are unsatisfactory (Ikinger, 1998; Richter, 2001). This led to Richter (2002) questioning its place within the Tolk-Sprenge subdivision of the Lyngby culture (*sensu* Taute 1968), comparable with the Segebro-Bromme culture in Scandinavia: despite the presence of backed form variants, these are thought to occur (in their minority) within the Tolk-Sprenge (see Taute 1968 and Ikinger 1998). However, with a distinct chronometric framework lacking (similarly to Bienenbüttel), and as a result of mechanical taphonomic processes at Häcklingen, it is unknown whether the Final Palaeolithic material represents one or multiple occupation events.  

Despite cross-contamination in layers, the typologically perspective awkward admixture of tanged and backed point variants at Häcklingen are particularly relevant to the robustness of these artefacts, and the combination of them as archaeological taxonomic units. 

For further details on the site of Häcklingen FStNr. 19 please refer to Richter (2002).  

[Figure 1: Select Backed and Tanged Points from Häcklingen FStNr. 19 and Bienenbüttel FStNr. 15 (redrawn after Richter 2001 and Richter 2002)]  


## Techno-typological Classifications and Points of Contention

The sites of Häcklingen and Bienenbüttel represent just two of a number of contexts, with a lack of chronometric dating, and known to feature a wealth of different named artefact units commonly thought to represent specific taxonomic groups. Others include XXXXXXX (FELIX ADD HERE). (ADD HERE WHY ITS IMPORTANT THAT THIS IS CLARIFIED)


# Materials and Methods

To better understand the technological positioning of Häcklingen and Bienenbüttel, an empirical quantitative and qualitative analysis is performed on technological and morphological variables on a number of previously classified Federmesser and Bromme material. For this analysis, the following contexts (in addition to Häcklingen and Bienenbüttel) are considered (see Figure 1):  

Previously classified Federmesser material

3)  **Rietberg (1/2/5/Other/Spoil)** (North Rhine-Westphalia, Germany): (ADD HERE)

4)  **Rothenkirchen** (Hesse, Germany): (ADD HERE)

Previously classified Bromme material

5)  **Brümmerhof (FStNr. 16)** (Lower Saxony, Germany): (ADD HERE)

6)  **Højgård** (Zealand, Denmark): (ADD HERE)

7)  **Sassenholz (FStNr. 78/82)** (Lower Saxony, Germany): (ADD HERE)

8)  **Skovmosen I** (Zealand, Denmark): A Final Palaeolithic assemblage comprising of tanged points, scrapers, burins, and primary reduction products including blades and cores (Boye, 2006; Hilgart, 2003; Eggers-Kaas et al. 2019). Can be traditionally placed within the Bromme culture (Hilgart, 2003; Boye, 2006; Brinch Petersen, 2009), however caution has been noted in its classification (Eggers-Kaas et al. 2019).

9)  **Søvind** (Jutland, Denmark): (ADD HERE)

10) **Segebro** (Malmö, Sweden): (ADD HERE)

In understanding the underlying *chaînes opératoires*, and in adopting a common terminology for identifying technological signatures and characteristics, attributes originate from the Nordic Blade Technology Network (REFERENCE). Minor amendments, and the extension of categories in a small number of instances were necessary to reflect Final Palaeolithic variability e.g. dorsal scar directionality. First, technological and morphological characteristics are first examined through **visual and descriptive summaries of data**. The significance of differences in the distribution of continuous variables across assemblages were assessed using **pairwise Mann--Whitney** testing (with Bonferroni-corrected *p* values). A **Gini-Simpson** index is then employed to examine intrasite variability while a **Morisita-Horn diversity index** is used to examine inter-assemblage dissimilarity; past applications in Crema (2014), Maiorano et al. (2020) and Leplongeon et al. (2020). **Multiple Correspondence Analysis (MCA)** is then employed for all categorical technological data, and for each data type (core, blade, backed point, tanged point); this will examine the degree of association between individual contexts and their technological make-up. **Principal Component Analysis (PCA)** is performed for all quantitative (morphometric) data, again for each data type; this will also examine individual contexts and their classification. PCA is analogous to MCA but for quantitative variables.  

(WHAT PHYLOGENETIC ANALYSIS SHOULD BE EMPLOYED: ML FOR QUANTITIVE VARIABLES AND PARSIMONY ANALYSES FOR QUALITATIVE?!)

All ordination methods in this analysis were conducted using the FactoMiner (Le et al. 2008), factoextra (Kassambara and Mundt, 2020) and the tidymodels packages (Kuhn et al. 2020), while diversity measures were calculated using the diverse (Guevara et al. 2016) and vegan (Oksanan et al. 2020) packages. A number of other packages were used to aid data wrangling, transformation and visualising, including rio (Chan et al. 2018), tidyverse (Wickham et al. 2019), gt (Iannone et al. 2020), broom (Robinson et al. 2020), tidytext (Silge and Robinson, 2016) and doBy (Højsgaard and Halekoh, 2020). All data, R code and supporting material (including figures) in this article can be found on the Open Science Framework (doi <https://osf.io/3h2rq/>)


```{r data import, echo = FALSE, message = FALSE}

data <- import("https://raw.githubusercontent.com/CSHoggard/-tech-data/master/Hoggard_and_Riede_2020.csv", na.strings=c(""), setclass = "tbl_df", encoding = "UTF-8")

cols <- c("CODE", 
          "CONTEXT", 
          "COUNTRY", 
          "CONDITION", 
          "ARTEFACT", 
          "SUB_CATEGORY", 
          "RECOVERY_METHOD", 
          "CLASSIFICATION", 
          "BP_ASSOCIATION", 
          "TP_ASSOCIATION", 
          "ABS_DATE_METHOD", 
          "RELAT_DATE_METHOD", 
          "RELAT_DATE_CHRONO", 
          "RAW_MAT", 
          "DORS_BLADE_PROF", 
          "BLADE_DET", 
          "BLADE_CURV", 
          "DORSAL_PATTERN", 
          "BULB_MORPH", 
          "CONUS_FORM", 
          "BUTT_MORPH", 
          "BUTT_PREP_1", 
          "BUTT_PREP_2", 
          "CORE_MORPH", 
          "PLAT_REJUV", 
          "CORE_METHOD", 
          "CORE_DIRECTIONALITY", 
          "CORE_TABLET_REJUV", 
          "CORE_FLAKE_REJUV", 
          "CORE_FRONT_REJUV", 
          "CORE_DIST_REJUV", 
          "CORE_SIDE_REJUV", 
          "BURINATION")

data[cols] <- lapply(data[cols], factor)

```

```{r maps, echo=FALSE, message=FALSE, fig.width = 8, fig.height = 8, fig.align = "center"}

world <- ne_countries(scale = "large", returnclass = "sf")

data.map <- data %>%
  distinct(CODE, LONGITUDE, LATITUDE, .keep_all = TRUE) %>%
  select(CODE, CONTEXT, LONGITUDE, LATITUDE, CLASSIFICATION) %>%
  mutate(CONTEXT = recode(CONTEXT, "Rietberg (Spoil)" = "Rietberg (1/2/5/Spoil/Other)",
                          "Sassenholz (FStNr. 82)" = "Sassenholz (FStNr. 78/82)"))

ggplot(data = world) +
  geom_sf(fill = "grey97") +
  geom_point(data = data.map, aes(x = LONGITUDE, y = LATITUDE, colour = CLASSIFICATION), size = 3) +
  coord_sf(xlim = c(6,15), ylim = c(50.5,57)) +
  labs(colour = "Classification", x = "Longitude", y = "Latitude") +
  geom_text_repel(data = data.map, aes(x = LONGITUDE, y = LATITUDE, label = CONTEXT, family = "Roboto Light")) + 
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) + 
   annotation_north_arrow(location = "tr", which_north = "true", 
        pad_x = unit(0.25, "in"), pad_y = unit(0.25, "in"),
        style = north_arrow_fancy_orienteering) +
  annotation_scale(location = "bl", width_hint = 0.4, text_family = "Roboto Light") +
  theme_minimal() + 
  theme(legend.position = "bottom",
        legend.title = element_text(family = "Roboto Light"),
        legend.text = element_text(family = "Roboto Light"),
        axis.title.x = element_text(family = "Roboto Light"),
        axis.title.y = element_text(family = "Roboto Light"),
        axis.text.x = element_text(family = "Roboto Light"),
        axis.text.y = element_text(family = "Roboto Light"))

```

```{r overview, echo = FALSE, message = FALSE}
data %>% 
  rename(Context = CONTEXT) %>%
  group_by(ARTEFACT) %>%
  count(Context) %>%
  pivot_wider(names_from = ARTEFACT, values_from = n) %>%
  gt() %>%
  tab_header(title = "Table 1a. Artefact Breakdown", subtitle = "Context") %>%
  opt_align_table_header(align = "left") 

data %>% 
  group_by(ARTEFACT) %>%
  count(CODE) %>%
  pivot_wider(names_from = ARTEFACT, values_from = n) %>%
  gt() %>%
  tab_header(title = "Table 1b. Artefact Breakdown", subtitle = "Code") %>%
  opt_align_table_header(align = "left")
```

```{r breakdown, echo = FALSE, message = FALSE}
data %>% 
  rename(Classification = CLASSIFICATION) %>%
  group_by(ARTEFACT) %>%
  count(Classification) %>%
  pivot_wider(names_from = ARTEFACT, values_from = n) %>%
  gt() %>%
  tab_header(title = "Table 2. Artefact Breakdown", subtitle = "Technology") %>%
  opt_align_table_header(align = "left")

```

```{r file creation, echo = FALSE, message = FALSE}
blade_data <- filter(data, ARTEFACT == "Blade") %>% droplevels()
core_data <- filter(data, ARTEFACT == "Core") %>% droplevels()
bp_data <- filter(data, ARTEFACT == "Backed Point") %>% droplevels()
tp_data <- filter(data, ARTEFACT == "Tanged Point") %>% droplevels()
```

# Technological attributes


The following attributes are recorded for all artefacts:

| Attribute | Variable(s)|
|-----------------------|----------------------------------------------------------------|
| *ID*                  | Catalogue Number / Identification Number|
| *CODE*                | Site Code|
| *CONTEXT*             | Archaeological Context|
| *COUTNRY*             | Country|
| *LONGITUDE*           | Longitude|
| *LATITUDE*            | Latitude|
| *RECOVERY METHOD*     | Recovery Method - 1: **Excavation**; 2: **Surface Collection**|
| *CLASSIFICATION*      | Taxonomic Unit e.g. Bromme|
| *BP_ASSOCIATION*      | Association of backed points - 1: **Yes**; 2: **No**|
| *TP_ASSOCIATION*      | Association of tanged points - 1: **Yes**; 2: **No**|
| *POINT_TYPE*          | Point type - 1: **Backed Point**; 2: **Tanged Point**|
| *NAMED_ARTEFACT_TYPE* | Classification e.g. Federmesser Point|
| *ABS_DATE_METHOD*     | Absolute dating method (if provided)|
| *ABS_DATE*            | Absolute date (as provided)|
| *ABS_DATE_STD*        | Absolute date standard deviation (as provided)|
| *RELAT_DATE_METHOD*   | Relative dating method (if provided)|
| *RELAT_DATE_CHRONO*   | Relative date (as provided)|
| *RAW_MAT*             | Raw material e.g. flint|

The following additional attributes are recorded for all **blades**:


| Attribute             | Variable(s)|
|-----------------------|----------------------------------------------------------------|
| *WEIGHT*              | Weight (g)|                                                                   
| *LENGTH*              | Length (mm)|                                                                  
| *WIDTH*               | Width (mm)|                                                                   
| *THICKNESS*           | Thickness (mm)|                                                               
| *ELONGATION*          | Elongation Index (Length/Width)|                                              
| *PLAT_DEPTH*          | Platform Depth (mm)|                                                          
| *DORS_BLADE_PROF*     | Dorsal blade characterisation - 1: **Full Cortical Dorsal Face (FC)**; 2: **Two Dorsal Faces - One Cortex (TDOC)**; 3: **Three Dorsal Faces - One Cortex (THDOC)**; 4: **Two Dorsal Faces - No Cortex (TDNC)**; 5: **Three Dorsal Faces - No Cortex (THDNC)**; 6: **Multiple Dorsal Faces (MDF)**; 7: **Bilateral Crested Blade (BCB)**; 8: **Crested Blade - One Flaked And One Uncortical (CBOU)**; 9: **Crested Blade - Three Flaked Faces (CBTHF)**; 10: **Crested Blade - One Flaked And One Cortical (CBOFOC)**; 11: **Crested Blade - Flaked And Trimmed (CBFT)**|
| *BLADE_DET*           | Blade determination - 1: **Ideal (ID)**; 2: **Feathered (FE)**; 3: **Plunged (PL)**; 4: **Hinged (HI)**       
| *BLADE_CURV*          | Blade curvature - 1: **Straight (ST)**; 2: **Distal (DI)**; 3: **Even (EV)**; 4: **Ventral 'Belly' (VB)**     
| *DORSAL_PATTERN*      | Dorsal scar pattern - 1: **Centripetal (CE)**; 2: **3-Way Centripetal (TWC)**; 3: **Bidirectional (BI)**; 4: **Convergent (CON)**; 5: **Convergent And Bidirectional (CONBI)**; 6: **Convergent And Perpendicular (CONPE)**; 7: **Double Perpendicular (DP)**; 8: **Straight And Perpendicular (SAP)**; 9: **Unidirectional (UNI)**; 10: **Undetermined (UND)**              
| *BULB_MORPH*          | Bulb and lip characteristics - 1: **Bulb Formation (BF)**; 2: **Pronounced Bulb Formation (PBF)**; 3: **Bulb And Lip Formation (BLF)**; 4: **Lip Formation (LF)**; 5: **Pronounced Lip Formation (PLF)**; 6: **Diffused Bulb (DB)**; 7: **No Bulb Or Lip (NBOL)**|        
| *CONUS_FORM*    | Cone Formation - 1: **No Formation (NFO)**; 2: **Ring Crack On Butt (RCB)**; 3: **Ring Crack And Ventral Fissures (RCVF)**; 4: **Detached Bulb (DB)**|                                  
| *BUTT_MORPH*          | Butt morphology - 1: **Large And Thick Butt (LTB)**; 2: **Large Oval Butt (LOB)**; 3: **Thin Oval Butt (TOB)**; 4: **Small Thick Butt (STB)**; 5: **Small Butt (SB)**; 6: **Punctiform Butt (PUNB)**; 7: **Broken/Absent Butt (BAB)**      
| *BUTT_PREP_1*         | Butt preparation #1 - 1: **Plain (PLA)**; 2: **Facetted With Two Scars (FTS)**; 3: **Facetted With Greater Than Two Scars (FGTTS)**; 4: **Broken (B)**|                       
| *BUTT_PREP_2*         | Butt preparation #2 - 1: **Cortical Unprepared (CU)**; 2: **Non-Cortical Unprepared (NCU)**; 3: **Dorsal Trimming (DT)**; 4: **Dorsal Abrasion (DA)**; 5: **Dorsal Abrasion And Grinding (DAG)**; 6; **Dorsal Abrasion And Trimming (DATR)**; 7: **Dorsal Abrasion, Trimming And Grinding (DATG)**; 8: **Broken (B)**|                                                                                                                                                                                            

Note: Blades are here defined as any previously assigned or reclassified material with an elongation index of 2:1, exhibits parallel lateral edges and appears to be derived from a scheme of stereotyped elongation production.

The following additional attributes are recorded for all **cores**:

| Attribute             | Variable(s)|
|-----------------------|----------------------------------------------------------------|
| *WEIGHT*              | Weight (g)|                                                                   
| *CORE_LENGTH*         | Core length (mm) as oriented on morphological axis (max length) and flaking surface (most blade removals)|
| *CORE_WIDTH*          | Core width (mm) as oriented on morphological axis (max length) and flaking surface (most blade removals)|
| *CORE_BREADTH*        | Core breadth (mm) as oriented on morphological axis (max length) and flaking surface (most blade removals)|
| *CORE_MORPH*          | Platform count: 1: **One Platform (OP)**; 2: **Two Platforms (TP)**|          
| *PLAT_REJUV*          | Platform description: 1: **Single Smooth (SS)**; 2: **Double Smooth (DS)**; 3: **Single Facetted/Flaked (SF)**; 4: **Double Facetted/Flaked (DF)**; 5: **Single Systematic (SSY)**; 6: **Double Systematic (DSY)**; 7: **Double Smooth And Facetted (DSF)**; 8: **Double Smooth And Systematic (DSS)**; 9: **Double Facetted And Systematic (DFSY)**|
| *CORE_METHOD*         | Core exploitation method - 1: **Semi-Rotating (SRO)**; 2: **Full-Rotating (FURO)**; 3: **Frontal (FRO)**; 4: **Facial (FAC)**; 5: **Multi-Facial (MFAC)**|
| *CORE_DIRECTIONALITY* | Scar directionality - 1: **Unidirectional (CUNI)**; 2: **Bidirectional (CBI)**; 3: **Mixed (CM)**|
| *CORE_TABLET_REJUV*   | Evidence for core tablet removals -- 1) **Yes (Y)**; 2) **No (N)**|           
| *CORE_FLAKE_REJUV*    | Evidence for preparatory flake rejuvenation -- 1) **Yes (Y)**; 2) **No (N)**|
| *CORE_FRONT_REJUV*    | Evidence for core frontal rejuvenation -- 1) **Yes (Y)**; 2) **No (N)**|
| *CORE_DIST_REJUV*     | Evidence for core distal rejuvenation -- 1) **Yes (Y); 2) No (N)**|
| *CORE_SIDE_REJUV*     | Evidence for core lateral rejuvenation -- 1) **Yes (Y); 2) No (N)**|

Note: Blade cores are here defined as material which exhibits the production of stereotyped elongated material around the core's circumference.

The following additional attributes are recorded for all **tanged and backed points**:

| Attribute             | Variable(s)|
|-----------------------|----------------------------------------------------------------|
| *WEIGHT*              | Weight (g)|
| *LENGTH*              | Length (mm)|
| *WIDTH*               | Width (mm)|
| *THICKNESS*           | Thickness (mm)|
| *ELONGATION*          | Elongation Index (Length/Width)|
| *TANG_ORIENTATION*    | Orientation of tang or base - 1) **Proximal**; 2) **Distal**; 3) **Lateral**| 
| *TCSA*                | Maximum tip cross-sectional area (mm²) *sensu* Shea (2006)|
| *TCSP*                | Maximum tip cross-sectional perimeter (mm) *sensu* Shea (2006)|
| *BURINATION*          | Presence of burination -- 1) **Yes (Y)**; 2) **No (N)**|


# Analysis: Blades (exc. Crested Blades and Broken Pieces)

## Visual and Descriptive Summaries of Data

```{r blade clean and display, echo = FALSE, message = FALSE}

blade_data_clean <- blade_data %>%
  filter(!DORS_BLADE_PROF == "BCB",
         !DORS_BLADE_PROF == "CBOU",
         !DORS_BLADE_PROF == "CBTHF",
         !DORS_BLADE_PROF == "CBOFOC",
         !DORS_BLADE_PROF == "CBFT",
         !BUTT_MORPH == "BAB", 
         !BUTT_PREP_1 == "B", 
         !BUTT_PREP_2 == "B")

blade_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 3. Descriptive Statistics: Blade Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

blade_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 4. Descriptive Statistics: Blade Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

blade_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 5. Descriptive Statistics: Blade Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
blade_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 6. Descriptive Statistics: Blade Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

blade_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 7. Descriptive Statistics: Blade Thickness (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

blade_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(PLAT_DEPTH),2),
            max = round(max(PLAT_DEPTH),2),
            mean = round(mean(PLAT_DEPTH),2),
            med = round(median(PLAT_DEPTH),2),
            sd = round(sd(PLAT_DEPTH),2),
            var = round(var(PLAT_DEPTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 8. Descriptive Statistics: Blade Platform Depth (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

blade_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 9. Descriptive Statistics: Blade Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

blade_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 10. Descriptive Statistics: Blade Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

blade_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 11. Descriptive Statistics: Blade Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
blade_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 12. Descriptive Statistics: Blade Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

blade_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 13. Descriptive Statistics: Blade Thickness (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

blade_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(PLAT_DEPTH),2),
            max = round(max(PLAT_DEPTH),2),
            mean = round(mean(PLAT_DEPTH),2),
            med = round(median(PLAT_DEPTH),2),
            sd = round(sd(PLAT_DEPTH),2),
            var = round(var(PLAT_DEPTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 14. Descriptive Statistics: Blade Platform Depth (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

```

```{r blade summary 1, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}
ggplot(blade_data_clean, aes(CODE, WEIGHT, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Weight (g)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(blade_data_clean, aes(CODE, LENGTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(blade_data_clean, aes(CODE, WIDTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Width (mm)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(blade_data_clean, aes(CODE, ELONGATION, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Elongation (L/W)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(blade_data_clean, aes(CODE, THICKNESS, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Thickness (mm)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(blade_data_clean, aes(CODE, PLAT_DEPTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Platform Depth (mm)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")
```

```{r blade l w plot, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 6, fig.align = "center"}
ggplot(blade_data_clean, aes(WIDTH, LENGTH, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
         geom_point() +
         labs(x = "Blade Width (mm)", y = "Blade Length (mm)") +
  stat_ellipse(type = "norm", linetype = 2) +
  lims(x = c(0,65), y = c(0,130)) +
  theme_minimal() +
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  geom_abline(intercept = 0, slope = 2, colour = "grey50") +
  theme(legend.position = "bottom")

ggplot(blade_data_clean, aes(WIDTH, LENGTH, colour = CODE, shape = CODE)) +
         geom_point() +
         labs(x = "Blade Width (mm)", y = "Blade Length (mm)") +
  lims(x = c(0,65), y = c(0,130)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5,6,7,8,9)) + 
  geom_abline(intercept = 0, slope = 2, colour = "grey50") +
  theme(legend.position = "bottom")

```

```{r blade tests, echo = FALSE, message = FALSE, warning = FALSE}

pairwise.wilcox.test(blade_data_clean$WEIGHT, blade_data_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(blade_data_clean$LENGTH, blade_data_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(blade_data_clean$WIDTH, blade_data_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(blade_data_clean$ELONGATION, blade_data_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(blade_data_clean$THICKNESS, blade_data_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(blade_data_clean$PLAT_DEPTH, blade_data_clean$CODE, p.adj = "bonf", exact = FALSE)

pairwise.wilcox.test(blade_data_clean$WEIGHT, blade_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(blade_data_clean$LENGTH, blade_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(blade_data_clean$WIDTH, blade_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(blade_data_clean$ELONGATION, blade_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(blade_data_clean$THICKNESS, blade_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(blade_data_clean$PLAT_DEPTH, blade_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)

```

```{r blade summary 2, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}

ggplot(blade_data_clean, aes(fill = RAW_MAT, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Raw Material", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = DORS_BLADE_PROF, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Dorsal Blade Profile", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = BLADE_DET, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Blade Determination Profile", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = BLADE_CURV, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Blade Curvature", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = DORSAL_PATTERN, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Dorsal Pattern Profile", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = BULB_MORPH, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Bulb Morphology", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = BUTT_PREP_1, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Butt Preparation Method #1", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = BUTT_PREP_2, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Butt Preparation Method #2", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = RAW_MAT, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Raw Material", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = DORS_BLADE_PROF, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Dorsal Blade Profile", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = BLADE_DET, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Blade Determination Profile", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = BLADE_CURV, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Blade Curvature", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = DORSAL_PATTERN, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Dorsal Pattern Profile", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = BULB_MORPH, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Bulb Morphology", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = BUTT_PREP_1, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Butt Preparation Method #1", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_data_clean, aes(fill = BUTT_PREP_2, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Butt Preparation Method #2", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

```

## Gini-Simpson and Morisita-Horn diversity indices

```{r blade diversity 1, echo = FALSE, message = FALSE,  fig.width = 6, fig.height = 4, fig.align = "center"}

diversity_blade_prof <- diverse::diversity(table(blade_data_clean$CODE, blade_data_clean$DORS_BLADE_PROF), type = "gini-simpson", category_row = FALSE)
diversity_blade_det  <- diverse::diversity(table(blade_data_clean$CODE, blade_data_clean$BLADE_DET), type = "gini-simpson", category_row = FALSE)
diversity_blade_curv <- diverse::diversity(table(blade_data_clean$CODE, blade_data_clean$BLADE_CURV), type = "gini-simpson", category_row = FALSE)
diversity_blade_dp   <- diverse::diversity(table(blade_data_clean$CODE, blade_data_clean$DORSAL_PATTERN), type = "gini-simpson", category_row = FALSE)
diversity_blade_bm   <- diverse::diversity(table(blade_data_clean$CODE, blade_data_clean$BULB_MORPH), type = "gini-simpson", category_row = FALSE)
diversity_blade_cf   <- diverse::diversity(table(blade_data_clean$CODE, blade_data_clean$CONUS_FORM), type = "gini-simpson", category_row = FALSE)
diversity_blade_btm  <- diverse::diversity(table(blade_data_clean$CODE, blade_data_clean$BUTT_MORPH), type = "gini-simpson", category_row = FALSE)
diversity_blade_bp1  <- diverse::diversity(table(blade_data_clean$CODE, blade_data_clean$BUTT_PREP_1), type = "gini-simpson", category_row = FALSE)
diversity_blade_bp2  <- diverse::diversity(table(blade_data_clean$CODE, blade_data_clean$BUTT_PREP_2), type = "gini-simpson", category_row = FALSE)

diversity_blade_prof %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Dorsal Profile",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_det %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Blade Determination Profile",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_curv %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Blade Curvature",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_dp %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Dorsal Pattern",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_bm %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Bulb Morphology",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_cf %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Conus Formation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_btm %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Butt Morphology",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_bp1 %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Butt Preparation #1",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_bp2 %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Butt Preparation #2",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_prof_class <- diverse::diversity(table(blade_data_clean$CLASSIFICATION, blade_data_clean$DORS_BLADE_PROF), type = "gini-simpson", category_row = FALSE)
diversity_blade_det_class  <- diverse::diversity(table(blade_data_clean$CLASSIFICATION, blade_data_clean$BLADE_DET), type = "gini-simpson", category_row = FALSE)
diversity_blade_curv_class <- diverse::diversity(table(blade_data_clean$CLASSIFICATION, blade_data_clean$BLADE_CURV), type = "gini-simpson", category_row = FALSE)
diversity_blade_dp_class   <- diverse::diversity(table(blade_data_clean$CLASSIFICATION, blade_data_clean$DORSAL_PATTERN), type = "gini-simpson", category_row = FALSE)
diversity_blade_bm_class   <- diverse::diversity(table(blade_data_clean$CLASSIFICATION, blade_data_clean$BULB_MORPH), type = "gini-simpson", category_row = FALSE)
diversity_blade_cf_class   <- diverse::diversity(table(blade_data_clean$CLASSIFICATION, blade_data_clean$CONUS_FORM), type = "gini-simpson", category_row = FALSE)
diversity_blade_btm_class  <- diverse::diversity(table(blade_data_clean$CLASSIFICATION, blade_data_clean$BUTT_MORPH), type = "gini-simpson", category_row = FALSE)
diversity_blade_bp1_class  <- diverse::diversity(table(blade_data_clean$CLASSIFICATION, blade_data_clean$BUTT_PREP_1), type = "gini-simpson", category_row = FALSE)
diversity_blade_bp2_class  <- diverse::diversity(table(blade_data_clean$CLASSIFICATION, blade_data_clean$BUTT_PREP_2), type = "gini-simpson", category_row = FALSE)

diversity_blade_prof_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Dorsal Profile",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_det_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Blade Determination Profile",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_curv_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Blade Curvature",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_dp_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Dorsal Pattern",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_bm_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Bulb Morphology",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_cf_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Conus Formation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_btm_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Butt Morphology",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_bp1_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Butt Preparation #1",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_blade_bp2_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Butt Preparation #2",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

```

```{r blade diversity 2, echo = FALSE, message = FALSE, fig.height = 3, fig.align = "center"}
horn_blade_prof <- vegdist(table(blade_data_clean$CODE, blade_data_clean$DORS_BLADE_PROF), method = "horn", diag = TRUE, upper = TRUE)
horn_blade_det  <- vegdist(table(blade_data_clean$CODE, blade_data_clean$BLADE_DET), method = "horn", diag = TRUE, upper = TRUE)
horn_blade_curv <- vegdist(table(blade_data_clean$CODE, blade_data_clean$BLADE_CURV), method = "horn", diag = TRUE, upper = TRUE)
horn_blade_dp   <- vegdist(table(blade_data_clean$CODE, blade_data_clean$DORSAL_PATTERN), method = "horn", diag = TRUE, upper = TRUE)
horn_blade_bm   <- vegdist(table(blade_data_clean$CODE, blade_data_clean$BULB_MORPH), method = "horn", diag = TRUE, upper = TRUE)
horn_blade_cf   <- vegdist(table(blade_data_clean$CODE, blade_data_clean$CONUS_FORM), method = "horn", diag = TRUE, upper = TRUE)
horn_blade_btm  <- vegdist(table(blade_data_clean$CODE, blade_data_clean$BUTT_MORPH), method = "horn", diag = TRUE, upper = TRUE)
horn_blade_bp1  <- vegdist(table(blade_data_clean$CODE, blade_data_clean$BUTT_PREP_1), method = "horn", diag = TRUE, upper = TRUE)
horn_blade_bp2  <- vegdist(table(blade_data_clean$CODE, blade_data_clean$BUTT_PREP_2), method = "horn", diag = TRUE, upper = TRUE)

tidy(horn_blade_prof) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Dorsal Profile: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_blade_det) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Blade Determination Profile: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_blade_curv) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Blade Curvature: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_blade_dp) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Dorsal Pattern: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_blade_bm) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Bulb Morphology: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_blade_cf) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Conus Formation: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_blade_btm) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Butt Morphology: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_blade_bp1) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Butt Preparation #1: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_blade_bp2) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Butt Preparation #2: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))
```

## Principal Component Analysis (PCA)

Note: Crested blades and broken/indistinguishable butts and bulbs are omitted from this analysis.

```{r blade pca, echo = FALSE, message = FALSE}

blade_pca_data <- blade_data %>%
  filter(!DORS_BLADE_PROF == "BCB",
         !DORS_BLADE_PROF == "CBOU",
         !DORS_BLADE_PROF == "CBTHF",
         !DORS_BLADE_PROF == "CBOFOC",
         !DORS_BLADE_PROF == "CBFT",
         !BUTT_MORPH == "BAB", 
         !BUTT_PREP_1 == "B", 
         !BUTT_PREP_2 == "B") %>%
  select(CODE, CONTEXT, CLASSIFICATION, WEIGHT, LENGTH, WIDTH, ELONGATION, THICKNESS, PLAT_DEPTH)

blade_pca <- PCA(blade_pca_data, graph = FALSE, quali.sup = 1:3)

```

```{r blade pca table, echo = FALSE, message = FALSE, warning = FALSE}

as.tibble(get_eig(blade_pca), rownames = "rowname") %>%
  rename(variance = variance.percent,
         cumulative = cumulative.variance.percent)

```

```{r blade pca scree, echo = FALSE, message = FALSE, fig.width = 4, fig.height = 3, fig.align="center"}

fviz_screeplot(blade_pca, xlim = c(1,4.75), ylim = c(0,50), barfill = "grey50", barcolor = "grey50", geom = "bar") +
  labs(title = "", x = "Dimension") + 
  theme_minimal()

```

```{r blade pca contributions, echo = FALSE, message = FALSE, fig.align="center"}

blade_pca_var <- get_pca_var(blade_pca)

as_tibble(blade_pca_var$contrib, rownames = "rowname") %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5) %>%
  gather("Contribution", "n", 2:6) %>%
  ggplot(aes(n, rowname, fill = rowname)) +
  geom_col(show.legend = FALSE, width = 0.75) +
  labs(title = "Contribution of Variables", x = "Contribution (%)", y = "Variable") +
  scale_fill_grey() +
  facet_wrap(~Contribution, nrow = 1) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))
```

```{r blade pca graphs, echo = FALSE, message = FALSE, fig.width = 7, fig.height = 7, fig.align="center"}

fviz_pca_biplot(blade_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = blade_pca_data$CLASSIFICATION,
                palette = c("#466995", "grey80", "grey40", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Classification)",
       x = "Dimension 1 (49.16%)",
       y = "Dimension 2 (18.97%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(blade_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = blade_pca_data$CLASSIFICATION,
                palette = c("#466995", "grey80", "grey40", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Classification)",
       x = "Dimension 2 (18.97%)",
       y = "Dimension 3 (14.13%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(blade_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = blade_pca_data$CODE,
                palette = c("#466995", "grey80", "#56A3A6", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Context)",
       x = "Dimension 1 (49.16%)",
       y = "Dimension 2 (18.97%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(blade_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = blade_pca_data$CODE,
                palette = c("#466995", "grey80", "#56A3A6", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Classification)",
       x = "Dimension 2 (18.97%)",
       y = "Dimension 3 (14.13%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")
```

```{r blade pca boxplots, echo = FALSE, message = FALSE, fig.width = 10, fig.height = 7, fig.align="center"}

blade_pca_x <- cbind(blade_pca_data, blade_pca$ind$coord) %>% 
  select(CODE, CONTEXT, CLASSIFICATION, Dim.1, Dim.2, Dim.3, Dim.4, Dim.5) %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5)

blade_pca_x %>%
  gather("Value", "n", 4:8) %>%
  ggplot(aes(Value, n, fill = CLASSIFICATION)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

blade_pca_x %>%
  gather("Value", "n", 4:8) %>%
  ggplot(aes(Value, n, fill = CODE)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

```

## Multiple Correspondence Analysis (MCA)

Note: for MCA all crested blades and broken butts were omitted from the analysis.  

```{r blade mca, echo = FALSE, message = FALSE}

blade_mca_data <- blade_data %>%
  select(CODE, CONTEXT, CLASSIFICATION,DORS_BLADE_PROF, BLADE_DET, BLADE_CURV,
         DORSAL_PATTERN, BULB_MORPH, CONUS_FORM, BUTT_MORPH, BUTT_PREP_1, BUTT_PREP_2) %>%
  filter(!DORS_BLADE_PROF == "BCB",
         !DORS_BLADE_PROF == "CBOU",
         !DORS_BLADE_PROF == "CBTHF",
         !DORS_BLADE_PROF == "CBOFOC",
         !DORS_BLADE_PROF == "CBFT",
         !BUTT_MORPH == "BAB", 
         !BUTT_PREP_1 == "B", 
         !BUTT_PREP_2 == "B")

blade_mca <- MCA(blade_mca_data, graph = FALSE, quali.sup = 1:3)

```

```{r blade mca table, echo = FALSE, message = FALSE, warning = FALSE}

as.tibble(get_eig(blade_mca), rownames = "rowname") %>%
  rename(variance = variance.percent,
         cumulative = cumulative.variance.percent)

```

```{r blade mca scree, echo = FALSE, message = FALSE, fig.width = 4, fig.height = 3, fig.align="center"}

fviz_screeplot(blade_mca, ylim = c(0,50), barfill = "grey50", barcolor = "grey50", geom = "bar") +
  labs(title = "") + 
  theme_minimal()

```

```{r blade mca contributions, echo = FALSE, message = FALSE, fig.width = 8, fig.height = 9, fig.align="center"}

blade_mca_var <- get_mca_var(blade_mca)

as_tibble(blade_mca_var$contrib, rownames = "rowname") %>%
    rename(`Dimension 1` = `Dim 1`,
           `Dimension 2` = `Dim 2`,
           `Dimension 3` = `Dim 3`,
           `Dimension 4` = `Dim 4`,
           `Dimension 5` = `Dim 5`) %>%
  gather("Contribution", "n", 2:6) %>%
  ggplot(aes(n, rowname, fill = rowname)) +
  geom_col(show.legend = FALSE, width = 0.75) +
  labs(title = "Contribution of Variables", 
       x = "Contribution (%)", 
       y = "Variable",
       caption = "Refer to the attribute catalogue for specific terminology") +
  scale_fill_grey() +
  facet_wrap(~Contribution, nrow = 1) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))
```

Key acronyms for responses in *Axis 1*:  

*DB*: Diffused bulb (**Bulb Morph**)  
*NBOL*: No bulb or lip (**Bulb Morph**)  
*CU*: Cortical unprepared (**Butt Prep #2**)  
*PUNB*: Punctiform butt (**Butt Morph**)  
*MDF*: Multiple dorsal faces (**Dorsal Blade Prof**)  

Key acronyms for responses in *Axis 2*:  
*NBOL*: No bulb or lip (**Bulb Morph**)  
*DT*: Dorsal trimming (**Butt Prep #2**)  
*DB*: Diffused bulb (**Bulb Morph**)  
*VB*: Ventral 'belly' (**Blade Curv**)  
*FE*: Feathered (**Blade Determ**)  

```{r blade mca graphs, echo = FALSE, message = FALSE, fig.width = 7, fig.height = 7, fig.align="center"}

blade_mca_x <- cbind(blade_mca_data, blade_mca$ind$coord) %>% 
  select(CODE, CONTEXT, CLASSIFICATION, `Dim 1`, `Dim 2`, `Dim 3`, `Dim 4`, `Dim 5`) %>%
    rename(`Dimension 1` = `Dim 1`,
           `Dimension 2` = `Dim 2`,
           `Dimension 3` = `Dim 3`,
           `Dimension 4` = `Dim 4`,
           `Dimension 5` = `Dim 5`)

ggplot(blade_mca_x, aes(`Dimension 1`, `Dimension 2`, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
  geom_point() +
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  labs(title = "MCA 1 vs. MCA 2 (Classification)",
       x = "Dimension 1 (4.90%)",
       y = "Dimension 2 (4.31%)") +
  stat_ellipse() + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_mca_x, aes(`Dimension 2`, `Dimension 3`, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
  geom_point() +
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  labs(title = "MCA 2 vs. MCA 3 (Classification)",
       x = "Dimension 2 (4.31%)",
       y = "Dimension 3 (4.11%)") +
  stat_ellipse() + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")
  
ggplot(blade_mca_x, aes(`Dimension 1`, `Dimension 2`, colour = CODE, shape = CODE)) +
  geom_point() +
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5,6,7,8,9)) +
  labs(title = "MCA 1 vs. MCA 2 (Code)",
       x = "Dimension 1 (4.90%)",
       y = "Dimension 2 (4.31%)") +
  stat_ellipse() + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(blade_mca_x, aes(`Dimension 2`, `Dimension 3`, colour = CODE, shape = CODE)) +
  geom_point() +
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5,6,7,8,9)) +
  labs(title = "MCA 2 vs. MCA 3 (Code)",
       x = "Dimension 2 (4.31%)",
       y = "Dimension 3 (4.11%)") + 
  stat_ellipse() + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")


```

```{r blade mca boxplots, echo = FALSE, message = FALSE, fig.width = 10, fig.height = 7, fig.align= "center"}
blade_mca_x %>%
  gather("Value", "n", 4:8) %>%
  ggplot(aes(Value, n, fill = CLASSIFICATION)) +
  geom_violin() +
  labs(title = "MCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

blade_mca_x %>%
  gather("Value", "n", 4:8) %>%
  ggplot(aes(Value, n, fill = CODE)) +
  geom_violin() +
  labs(title = "MCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 
```

# Analysis: Cores

## Visual and Descriptive Summaries of Data

```{r core clean and display, echo = FALSE, message = FALSE}

core_data_clean <- core_data

core_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 15. Descriptive Statistics: Blade Core Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

core_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(CORE_LENGTH),2),
            max = round(max(CORE_LENGTH),2),
            mean = round(mean(CORE_LENGTH),2),
            med = round(median(CORE_LENGTH,2)),
            sd = round(sd(CORE_LENGTH),2),
            var = round(var(CORE_LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 16. Descriptive Statistics: Blade Core Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

core_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(CORE_WIDTH),2),
            max = round(max(CORE_WIDTH),2),
            mean = round(mean(CORE_WIDTH),2),
            med = round(median(CORE_WIDTH),2),
            sd = round(sd(CORE_WIDTH),2),
            var = round(var(CORE_WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 17. Descriptive Statistics: Blade Core Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
core_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(CORE_BREADTH),2),
            max = round(max(CORE_BREADTH),2),
            mean = round(mean(CORE_BREADTH),2),
            med = round(median(CORE_BREADTH),2),
            sd = round(sd(CORE_BREADTH),2),
            var = round(var(CORE_BREADTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 18. Descriptive Statistics: Blade Core Breadth (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

core_data_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(CORE_ELONGATION),2),
            max = round(max(CORE_ELONGATION),2),
            mean = round(mean(CORE_ELONGATION),2),
            med = round(median(CORE_ELONGATION),2),
            sd = round(sd(CORE_ELONGATION),2),
            var = round(var(CORE_ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 19. Descriptive Statistics: Blade Core Elongation (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

core_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 20. Descriptive Statistics: Blade Core Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

core_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(CORE_LENGTH),2),
            max = round(max(CORE_LENGTH),2),
            mean = round(mean(CORE_LENGTH),2),
            med = round(median(CORE_LENGTH,2)),
            sd = round(sd(CORE_LENGTH),2),
            var = round(var(CORE_LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 21. Descriptive Statistics: Blade Core Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

core_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(CORE_WIDTH),2),
            max = round(max(CORE_WIDTH),2),
            mean = round(mean(CORE_WIDTH),2),
            med = round(median(CORE_WIDTH),2),
            sd = round(sd(CORE_WIDTH),2),
            var = round(var(CORE_WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 22. Descriptive Statistics: Blade Core Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
core_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(CORE_BREADTH),2),
            max = round(max(CORE_BREADTH),2),
            mean = round(mean(CORE_BREADTH),2),
            med = round(median(CORE_BREADTH),2),
            sd = round(sd(CORE_BREADTH),2),
            var = round(var(CORE_BREADTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 23. Descriptive Statistics: Blade Core Breadth (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

core_data_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(CORE_ELONGATION),2),
            max = round(max(CORE_ELONGATION),2),
            mean = round(mean(CORE_ELONGATION),2),
            med = round(median(CORE_ELONGATION),2),
            sd = round(sd(CORE_ELONGATION),2),
            var = round(var(CORE_ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 24. Descriptive Statistics: Blade Core Elongation (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")


```

```{r core summary 1, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}
ggplot(core_data_clean, aes(CODE, WEIGHT, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Core Weight (g)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(core_data_clean, aes(CODE, CORE_LENGTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Core Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(core_data_clean, aes(CODE, CORE_WIDTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Core Width (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(core_data_clean, aes(CODE, CORE_BREADTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Core Breadth (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(core_data_clean, aes(CODE, CORE_ELONGATION, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Core Elongation Index", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")
```

```{r core l w plot, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 6, fig.align = "center"}
ggplot(core_data_clean, aes(CORE_WIDTH, CORE_LENGTH, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
         geom_point() +
         labs(x = "Core Width (mm)", y = "Core Length (mm)", fill = "Taxonomic Unit") +
  stat_ellipse(type = "norm", linetype = 2) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(legend.position = "bottom")

ggplot(core_data_clean, aes(CORE_WIDTH, CORE_LENGTH, colour = CODE, shape = CODE)) +
         geom_point() +
         labs(x = "Core Width (mm)", y = "Core Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  lims(x = c(0,90), y = c(0,180)) +
  scale_colour_manual(values = c("#466995", "#56A3A6", "grey80", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5,6,7,8,9)) + 
  theme(legend.position = "bottom")

```

```{r core tests, echo = FALSE, warning = FALSE, message = FALSE}

pairwise.wilcox.test(core_data_clean$WEIGHT, core_data_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(core_data_clean$CORE_LENGTH, core_data_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(core_data_clean$CORE_WIDTH, core_data_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(core_data_clean$CORE_BREADTH, core_data_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(core_data_clean$CORE_ELONGATION, core_data_clean$CODE, p.adj = "bonf", exact = FALSE)

pairwise.wilcox.test(core_data_clean$WEIGHT, core_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(core_data_clean$CORE_LENGTH, core_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(core_data_clean$CORE_WIDTH, core_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(core_data_clean$CORE_BREADTH, core_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(core_data_clean$CORE_ELONGATION, core_data_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)

```


```{r core summary 2, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}

ggplot(core_data_clean, aes(fill = RAW_MAT, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Raw Material", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_MORPH, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Morphology", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = PLAT_REJUV, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Platform Profile", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_METHOD, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Reduction Strategy", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_DIRECTIONALITY, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Directionality", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_TABLET_REJUV, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Tablet Rejuvenation", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_FLAKE_REJUV, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Preparatory Flake Rejuvenation", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_FRONT_REJUV, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Frontal Rejuvenation", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_DIST_REJUV, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Distal Rejuvenation", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_SIDE_REJUV, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Lateral Rejuvenation", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = RAW_MAT, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Raw Material", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_MORPH, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Morphology", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = PLAT_REJUV, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Platform Profile", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_METHOD, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Reduction Strategy", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_DIRECTIONALITY, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Directionality", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_TABLET_REJUV, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Tablet Rejuvenation", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_FLAKE_REJUV, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Preparatory Flake Rejuvenation", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_FRONT_REJUV, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Frontal Rejuvenation", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_DIST_REJUV, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Distal Rejuvenation", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_data_clean, aes(fill = CORE_SIDE_REJUV, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Core Lateral Rejuvenation", x = "", y = "Count", fill = "Attribute") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")
```

## Gini-Simpson and Morisita-Horn diversity indices

```{r core diversity 1, echo = FALSE, message = FALSE,  fig.width = 6, fig.height = 4, fig.align = "center"}

diversity_core_morph <- diverse::diversity(table(core_data_clean$CODE, core_data_clean$CORE_MORPH), type = "gini-simpson", category_row = FALSE)
diversity_core_pr  <- diverse::diversity(table(core_data_clean$CODE, core_data_clean$PLAT_REJUV), type = "gini-simpson", category_row = FALSE)
diversity_core_cm <- diverse::diversity(table(core_data_clean$CODE, core_data_clean$CORE_METHOD), type = "gini-simpson", category_row = FALSE)
diversity_core_cd   <- diverse::diversity(table(core_data_clean$CODE, core_data_clean$CORE_DIRECTIONALITY), type = "gini-simpson", category_row = FALSE)
diversity_core_ctr   <- diverse::diversity(table(core_data_clean$CODE, core_data_clean$CORE_TABLET_REJUV), type = "gini-simpson", category_row = FALSE)
diversity_core_flr   <- diverse::diversity(table(core_data_clean$CODE, core_data_clean$CORE_FLAKE_REJUV), type = "gini-simpson", category_row = FALSE)
diversity_core_frr   <- diverse::diversity(table(core_data_clean$CODE, core_data_clean$CORE_FRONT_REJUV), type = "gini-simpson", category_row = FALSE)
diversity_core_cdr   <- diverse::diversity(table(core_data_clean$CODE, core_data_clean$CORE_DIST_REJUV), type = "gini-simpson", category_row = FALSE)
diversity_core_csr  <- diverse::diversity(table(core_data_clean$CODE, core_data_clean$CORE_SIDE_REJUV), type = "gini-simpson", category_row = FALSE)


diversity_core_morph %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Morphology",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_pr %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Platform Profile",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_cm %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Reduction Strategy",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_cd %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Directionality",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_ctr %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Tablet Rejuvenation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_flr %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Preparatory Flake Rejuvenation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_frr %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Frontal Rejuvenation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_cdr %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Distal Rejuvenation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_csr %>% 
  rownames_to_column(var = "Context") %>%
  mutate("Context" = as.factor(Context)) %>%
  ggplot(aes(Context, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Lateral Rejuvenation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_morph_class <- diverse::diversity(table(core_data_clean$CLASSIFICATION, core_data_clean$CORE_MORPH), type = "gini-simpson", category_row = FALSE)
diversity_core_pr_class  <- diverse::diversity(table(core_data_clean$CLASSIFICATION, core_data_clean$PLAT_REJUV), type = "gini-simpson", category_row = FALSE)
diversity_core_cm_class <- diverse::diversity(table(core_data_clean$CLASSIFICATION, core_data_clean$CORE_METHOD), type = "gini-simpson", category_row = FALSE)
diversity_core_cd_class   <- diverse::diversity(table(core_data_clean$CLASSIFICATION, core_data_clean$CORE_DIRECTIONALITY), type = "gini-simpson", category_row = FALSE)
diversity_core_ctr_class   <- diverse::diversity(table(core_data_clean$CLASSIFICATION, core_data_clean$CORE_TABLET_REJUV), type = "gini-simpson", category_row = FALSE)
diversity_core_flr_class   <- diverse::diversity(table(core_data_clean$CLASSIFICATION, core_data_clean$CORE_FLAKE_REJUV), type = "gini-simpson", category_row = FALSE)
diversity_core_frr_class   <- diverse::diversity(table(core_data_clean$CLASSIFICATION, core_data_clean$CORE_FRONT_REJUV), type = "gini-simpson", category_row = FALSE)
diversity_core_cdr_class   <- diverse::diversity(table(core_data_clean$CLASSIFICATION, core_data_clean$CORE_DIST_REJUV), type = "gini-simpson", category_row = FALSE)
diversity_core_csr_class  <- diverse::diversity(table(core_data_clean$CLASSIFICATION, core_data_clean$CORE_SIDE_REJUV), type = "gini-simpson", category_row = FALSE)

diversity_core_morph_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Morphology",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_pr_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Platform Profile",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_cm_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Reduction Strategy",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_cd_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Directionality",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_ctr_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Tablet Rejuvenation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_flr_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Preparatory Rejuvenation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_frr_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Frontal Rejuvenation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_cdr_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Distal Rejuvenation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

diversity_core_csr_class %>% 
  rownames_to_column(var = "Classification") %>%
  mutate("Classification" = as.factor(Classification)) %>%
  ggplot(aes(Classification, gini.simpson)) +
  geom_point(size = 4) +
  geom_line(aes(group = 1)) +
  ylim(0,1) + 
  labs(title = "Core Lateral Rejuvenation",x = "", y = "Gini Simpson Index") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18))

```


```{r core diversity 2, echo = FALSE, message = FALSE, fig.height = 3, fig.align = "center"}
horn_core_morph <- vegdist(table(core_data_clean$CODE, core_data_clean$CORE_MORPH), method = "horn", diag = TRUE, upper = TRUE)
horn_core_pr  <- vegdist(table(core_data_clean$CODE, core_data_clean$PLAT_REJUV), method = "horn", diag = TRUE, upper = TRUE)
horn_core_cm <- vegdist(table(core_data_clean$CODE, core_data_clean$CORE_METHOD), method = "horn", diag = TRUE, upper = TRUE)
horn_core_cd   <- vegdist(table(core_data_clean$CODE, core_data_clean$CORE_DIRECTIONALITY), method = "horn", diag = TRUE, upper = TRUE)
horn_core_tr   <- vegdist(table(core_data_clean$CODE, core_data_clean$CORE_TABLET_REJUV), method = "horn", diag = TRUE, upper = TRUE)
horn_core_fl   <- vegdist(table(core_data_clean$CODE, core_data_clean$CORE_FLAKE_REJUV), method = "horn", diag = TRUE, upper = TRUE)
horn_core_fr   <- vegdist(table(core_data_clean$CODE, core_data_clean$CORE_FRONT_REJUV), method = "horn", diag = TRUE, upper = TRUE)
horn_core_dr   <- vegdist(table(core_data_clean$CODE, core_data_clean$CORE_DIST_REJUV), method = "horn", diag = TRUE, upper = TRUE)
horn_core_sr  <- vegdist(table(core_data_clean$CODE, core_data_clean$CORE_SIDE_REJUV), method = "horn", diag = TRUE, upper = TRUE)


tidy(horn_core_morph) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Core Morphology: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_core_pr) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Core Platform Profile: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_core_cm) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Core Reduction Strategy: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_core_cd) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Core Directionality: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_core_tr) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Core Tablet Rejuvenation: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_core_fl) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Preparatory Flake Rejuvenation: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_core_fr) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Frontal Rejuvenation: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_core_dr) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Core Distal Rejuvenation: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

tidy(horn_core_sr) %>%
  mutate("Context_1" = item1,
         "Context_2" = item2) %>%
  ggplot(aes(Context_1, Context_2, fill = distance)) + 
  geom_tile() +
  geom_text(aes(label = round(distance, 1))) +
  labs(title = "Core Lateral Rejuvenation: Morisita-Horn Index", x = "", y = "", fill = "Distance") +
  scale_fill_gradient(low="white", high="grey40") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16))

```

## Principal Component Analysis (PCA)

```{r core pca, echo = FALSE, message = FALSE}

core_pca_data <- core_data  %>%
  select(CODE, CONTEXT, CLASSIFICATION, WEIGHT, CORE_LENGTH, CORE_WIDTH, CORE_ELONGATION, CORE_BREADTH)

core_pca <- PCA(core_pca_data, graph = FALSE, quali.sup = 1:3)

```

```{r core pca table, echo = FALSE, message = FALSE, warning = FALSE}

as.tibble(get_eig(core_pca), rownames = "rowname") %>%
  rename(variance = variance.percent,
         cumulative = cumulative.variance.percent)

```

```{r core pca scree, echo = FALSE, message = FALSE, fig.width = 4, fig.height = 3, fig.align="center"}

fviz_screeplot(core_pca, ylim = c(0,50), barfill = "grey50", barcolor = "grey50", geom = "bar") +
  labs(title = "", x = "Dimension") + 
  theme_minimal()

```

```{r core pca contribution, echo = FALSE, message = FALSE, fig.align="center"}

core_pca_var <- get_pca_var(core_pca)

as_tibble(core_pca_var$contrib, rownames = "rowname") %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5) %>%
  gather("Contribution", "n", 2:6) %>%
  ggplot(aes(n, rowname, fill = rowname)) +
  geom_col(show.legend = FALSE, width = 0.75) +
  labs(title = "Contribution of Variables", x = "Contribution (%)", y = "Variable") +
  scale_fill_grey() +
  facet_wrap(~Contribution, nrow = 1) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))
```


```{r core pca graphs, echo = FALSE, message = FALSE, fig.width = 7, fig.height = 7, fig.align="center"}

fviz_pca_biplot(core_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = core_pca_data$CLASSIFICATION,
                palette = c("#466995", "grey40", "grey80", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Classification)",
       x = "Dimension 1 (56.67%)",
       y = "Dimension 2 (26.98%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(core_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = core_pca_data$CLASSIFICATION,
                palette = c("#466995", "grey40", "grey80", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Classification)",
       x = "Dimension 2 (26.98%)",
       y = "Dimension 3 (9.53%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(core_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = core_pca_data$CODE,
                palette = c("#466995", "#56A3A6", "grey80", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Context)",
       x = "Dimension 1 (56.67%)",
       y = "Dimension 2 (26.98%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(core_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = core_pca_data$CODE,
                palette = c("#466995", "#56A3A6", "grey80", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Context)",
       x = "Dimension 2 (26.98%)",
       y = "Dimension 3 (9.53%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

```

```{r core pca boxplots, echo = FALSE, message = FALSE, fig.width = 10, fig.height = 7, fig.align="center"}

core_pca_x <- cbind(core_pca_data, core_pca$ind$coord) %>% 
  select(CODE, CONTEXT, CLASSIFICATION, Dim.1, Dim.2, Dim.3, Dim.4, Dim.5) %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5)

core_pca_x %>%
  gather("Value", "n", 4:8) %>%
  ggplot(aes(Value, n, fill = CLASSIFICATION)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

core_pca_x %>%
  gather("Value", "n", 4:8) %>%
  ggplot(aes(Value, n, fill = CODE)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "#56A3A6", "grey80", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

```

## Multiple Correspondence Analysis (MCA)

```{r core mca, echo = FALSE, message = FALSE}

core_mca_data <- core_data %>% select(CODE, CONTEXT, CLASSIFICATION, CORE_MORPH, PLAT_REJUV, CORE_METHOD, CORE_DIRECTIONALITY, CORE_TABLET_REJUV, CORE_FLAKE_REJUV, CORE_FRONT_REJUV, CORE_DIST_REJUV, CORE_SIDE_REJUV)

core_mca <- MCA(core_mca_data, graph = FALSE, quali.sup = 1:3)

```

```{r core mca table, echo = FALSE, message = FALSE, warning = FALSE}

as.tibble(get_eig(core_mca), rownames = "rowname") %>%
  rename(variance = variance.percent,
         cumulative = cumulative.variance.percent)

```

```{r core mca scree, echo = FALSE, message = FALSE, fig.width = 4, fig.height = 3, fig.align="center"}

fviz_screeplot(core_mca, ylim = c(0,50), barfill = "grey50", barcolor = "grey50", geom = "bar") +
  labs(title = "") + 
  theme_minimal()

```

```{r core mca contributions, echo = FALSE, message = FALSE, fig.width = 8, fig.height = 9, fig.align="center"}

core_mca_var <- get_mca_var(core_mca)

as_tibble(core_mca_var$contrib, rownames = "rowname") %>%
    rename(`Dimension 1` = `Dim 1`,
           `Dimension 2` = `Dim 2`,
           `Dimension 3` = `Dim 3`,
           `Dimension 4` = `Dim 4`,
           `Dimension 5` = `Dim 5`) %>%
  gather("Contribution", "n", 2:6) %>%
  ggplot(aes(n, rowname, fill = rowname)) +
  geom_col(show.legend = FALSE, width = 0.75) +
  labs(title = "Contribution of Variables", 
       x = "Contribution (%)", 
       y = "Variable",
       caption = "Refer to the attribute catalogue for specific terminology") +
  scale_fill_grey() +
  facet_wrap(~Contribution, nrow = 1) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))
```

Key acronyms for responses in *Axis 1*:  

*CORE_FLAKE_REJUV_Y*: Preparatory flake rejuvenation: positive response (**Core Flake Rejuv**)  
*CORE_SIDE_REJUV_Y*:  Lateral rejuvenation: positive response (**Core Side Rejuv**)  
*CORE_TABLET_REJUV_Y*: Core tabletting: positive response (**Core Tablet Rejuv**)  
*DSF*: Double smooth and facetted core platforms (**Plat Rejuv**)  
*CBI*: Bidirectional blade scars (**Core Directionality**)  

Key acronyms for responses in *Axis 2*:  
*TP*: Two platforms (**Core Morph**)  
*SF*: Single facetted and/or flaked platforms (**Plat Rejuv**)  
*DS*: Double smooth platforms (**Plat Rejuv**)  
*OP*: One platform (**Core Morph**)  
*CORE_TABLET_REJUV_Y*: Core tabletting: positive response(**Core Tablet Rejuv**)  

```{r core mca graphs, echo = FALSE, message = FALSE, fig.width = 7, fig.height = 7, fig.align="center"}

core_mca_x <- cbind(core_mca_data, core_mca$ind$coord) %>% 
  select(CODE, CONTEXT, CLASSIFICATION, `Dim 1`, `Dim 2`, `Dim 3`, `Dim 4`, `Dim 5`) %>%
    rename(`Dimension 1` = `Dim 1`,
           `Dimension 2` = `Dim 2`,
           `Dimension 3` = `Dim 3`,
           `Dimension 4` = `Dim 4`,
           `Dimension 5` = `Dim 5`)

ggplot(core_mca_x, aes(`Dimension 1`, `Dimension 2`, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
  geom_point() +
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  labs(title = "MCA 1 vs. MCA 2 (Classification)",
       x = "Dimension 1 (14.53%)",
       y = "Dimension 2 (12.36%)") +
  stat_ellipse() + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_mca_x, aes(`Dimension 2`, `Dimension 3`, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
  geom_point() +
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  labs(title = "MCA 2 vs. MCA 3 (Classification)",
       x = "Dimension 2 (12.36%)",
       y = "Dimension 3 (9.51%)") +
  stat_ellipse() + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")
  
ggplot(core_mca_x, aes(`Dimension 1`, `Dimension 2`, colour = CODE, shape = CODE)) +
  geom_point() +
  scale_colour_manual(values = c("#466995", "#56A3A6", "grey80", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5,6,7,8,9)) +
  labs(title = "MCA 1 vs. MCA 2 (Code)",
       x = "Dimension 1 (14.53%)",
       y = "Dimension 2 (12.36%)") +
  stat_ellipse() + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(core_mca_x, aes(`Dimension 2`, `Dimension 3`, colour = CODE, shape = CODE)) +
  geom_point() +
  scale_colour_manual(values = c("#466995", "#56A3A6", "grey80", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5,6,7,8,9)) +
  labs(title = "MCA 2 vs. MCA 3 (Code)",
       x = "Dimension 2 (12.36%)",
       y = "Dimension 3 (9.51%)") + 
  stat_ellipse() + 
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

```

```{r core mca boxplots, echo = FALSE, message = FALSE, fig.width = 10, fig.height = 7, fig.align= "center"}
core_mca_x %>%
  gather("Value", "n", 4:8) %>%
  ggplot(aes(Value, n, fill = CLASSIFICATION)) +
  geom_violin() +
  labs(title = "MCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

core_mca_x %>%
  gather("Value", "n", 4:8) %>%
  ggplot(aes(Value, n, fill = CODE)) +
  geom_violin() +
  labs(title = "MCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "#56A3A6", "grey80", "grey60", "grey50", "grey40", "grey30", "grey20", "grey10")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 
```

# Analysis: Backed Points (exc. burinated pieces)

Note: Backed pieces categorised as "Creswellian/Penknife Point", "Kent Point" and "Penknife Point" have, for the purposes of this analysis been redefined as "Creswellian Point".

## Visual and Descriptive Summaries of Data

```{r bp excb clean and display, echo = FALSE, message = FALSE}

bp_data_excb_clean <- bp_data %>%
  filter(!BURINATION == "Y") %>%
  mutate(SUB_CATEGORY = fct_recode(SUB_CATEGORY, "Creswellian Point" = "Creswellian/Penknife Point"),
         SUB_CATEGORY = fct_recode(SUB_CATEGORY, "Creswellian Point" = "Penknife Point"),
         SUB_CATEGORY = fct_recode(SUB_CATEGORY, "Creswellian Point" = "Kent Point")) %>%
  rename(BASE_LOCATION = TANG_ORIENTATION)

bp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 25. Descriptive Statistics: Backed Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 26. Descriptive Statistics: Backed Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 27. Descriptive Statistics: Backed Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
bp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 28. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 29. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 30. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 31. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 32. Descriptive Statistics: Backed Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 33. Descriptive Statistics: Backed Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 34. Descriptive Statistics: Backed Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
bp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 35. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 36. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 37. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 38. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 39. Descriptive Statistics: Backed Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 40. Descriptive Statistics: Backed Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 41. Descriptive Statistics: Backed Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
bp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 42. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 43. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 44. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 45. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

```

```{r bp excb summary 1, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}

ggplot(bp_data_excb_clean, aes(CODE, WEIGHT, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Weight (g)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(CODE, LENGTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(CODE, WIDTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Width (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(CODE, ELONGATION, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Elongation (L/W)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(CODE, THICKNESS, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Thickness (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(CODE, TCSA, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSA)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(CODE, TCSP, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSP)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(SUB_CATEGORY, WEIGHT)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Weight (g)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(SUB_CATEGORY, LENGTH)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(SUB_CATEGORY, WIDTH)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Width (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(SUB_CATEGORY, ELONGATION)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Elongation (L/W)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(SUB_CATEGORY, THICKNESS)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Thickness (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(SUB_CATEGORY, TCSA)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSA)", fill = "Taxonomic Unit") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(SUB_CATEGORY, TCSP)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSP)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")
```

```{r bp excb l w plot, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 6, fig.align = "center"}

ggplot(bp_data_excb_clean, aes(WIDTH, LENGTH, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
         geom_point() +
         labs(x = "Backed Point Width (mm)", y = "Backed Point Length (mm)") +
  stat_ellipse(type = "norm", linetype = 2) +
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(WIDTH, LENGTH, colour = CODE, shape = CODE)) +
         geom_point() +
         labs(x = "Backed Point Width (mm)", y = "Backed Point Length (mm)") +
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20")) +
  scale_shape_manual(values = c(1,2,3,4,5,6)) +
  theme(legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(WIDTH, LENGTH, colour = CODE, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "Backed Point Width (mm)", y = "Backed Point Length (mm)") + 
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20")) +
  scale_shape_manual(values = c(1,2,3))

ggplot(bp_data_excb_clean, aes(WIDTH, LENGTH, colour = CLASSIFICATION, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "Backed Point Width (mm)", y = "Backed Point Length (mm)") + 
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() +
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  scale_shape_manual(values = c(1,2,3))

```

```{r bp excb tcsa tcsp plot, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 6, fig.align = "center"}

ggplot(bp_data_excb_clean, aes(TCSA, TCSP, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") +
  stat_ellipse(type = "norm", linetype = 2) +
  lims(x = c(0,100), y = c(0,50)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(TCSA, TCSP, colour = CODE, shape = CODE)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") +
  lims(x = c(0,100), y = c(0,50)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20")) +
  scale_shape_manual(values = c(1,2,3,4,5,6)) +
  theme(legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(TCSA, TCSP, colour = CODE, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") + 
  lims(x = c(0,100), y = c(0,50)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20")) +
  scale_shape_manual(values = c(1,2,3))

ggplot(bp_data_excb_clean, aes(TCSA, TCSP, colour = CLASSIFICATION, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") + 
  lims(x = c(0,100), y = c(0,50)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  scale_shape_manual(values = c(1,2,3))

```

```{r bp excb tests, echo = FALSE, message = FALSE, warning = FALSE}

pairwise.wilcox.test(bp_data_excb_clean$WEIGHT, bp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$LENGTH, bp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$WIDTH, bp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$ELONGATION, bp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$THICKNESS, bp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$TCSA, bp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$TCSP, bp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)

pairwise.wilcox.test(bp_data_excb_clean$WEIGHT, bp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$LENGTH, bp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$WIDTH, bp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$ELONGATION, bp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$THICKNESS, bp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$TCSA, bp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$TCSP, bp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)

pairwise.wilcox.test(bp_data_excb_clean$WEIGHT, bp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$LENGTH, bp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$WIDTH, bp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$ELONGATION, bp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$THICKNESS, bp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$TCSA, bp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_excb_clean$TCSP, bp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)

```

```{r bp excb summary 2, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}

ggplot(bp_data_excb_clean, aes(fill = BASE_LOCATION, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Base Location", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(fill = BASE_LOCATION, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Base Location", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(bp_data_excb_clean, aes(fill = BASE_LOCATION, x = SUB_CATEGORY)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Base Location", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

```

## Principal Component Analysis (PCA)

```{r bp excb pca, echo = FALSE, message = FALSE}

bp_excb_pca_data <- bp_data_excb_clean %>%
  select(CODE, CONTEXT, CLASSIFICATION, SUB_CATEGORY, WEIGHT, LENGTH, WIDTH, ELONGATION, THICKNESS, TCSA, TCSP)

bp_excb_pca <- PCA(bp_excb_pca_data, graph = FALSE, quali.sup = 1:4)

```

```{r bp excb pca table, echo = FALSE, message = FALSE, warning = FALSE}

as.tibble(get_eig(bp_excb_pca), rownames = "rowname") %>%
  rename(variance = variance.percent,
         cumulative = cumulative.variance.percent)

```

```{r bp excb pca scree, echo = FALSE, message = FALSE, fig.width = 4, fig.height = 3, fig.align="center"}

fviz_screeplot(bp_excb_pca, ylim = c(0,50), barfill = "grey50", barcolor = "grey50", geom = "bar") +
  labs(title = "", x = "Dimension") + 
  theme_minimal()

```

```{r bp excb pca contributions, echo = FALSE, message = FALSE, fig.align="center"}

bp_excb_pca_var <- get_pca_var(bp_excb_pca)

as_tibble(bp_excb_pca_var$contrib, rownames = "rowname") %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5) %>%
  gather("Contribution", "n", 2:6) %>%
  ggplot(aes(n, rowname, fill = rowname)) +
  geom_col(show.legend = FALSE, width = 0.75) +
  labs(title = "Contribution of Variables", x = "Contribution (%)", y = "Variable") +
  scale_fill_grey() +
  facet_wrap(~Contribution, nrow = 1) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))
```

```{r bp excb pca graphs, echo = FALSE, message = FALSE, fig.width = 7, fig.height = 7, fig.align="center"}

fviz_pca_biplot(bp_excb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = bp_excb_pca_data$CLASSIFICATION,
                palette = c("#466995", "grey40", "grey80", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Classification)",
       x = "Dimension 1 (62.35%)",
       y = "Dimension 2 (18.96%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(bp_excb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = bp_excb_pca_data$CLASSIFICATION,
                palette = c("#466995", "grey40", "grey80", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Classification)",
       x = "Dimension 2 (18.96%)",
       y = "Dimension 3 (10.10%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(bp_excb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = bp_excb_pca_data$CODE,
                palette = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Code)",
       x = "Dimension 1 (62.35%)",
       y = "Dimension 2 (18.96%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(bp_excb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = bp_excb_pca_data$CODE,
                palette = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Code)",
       x = "Dimension 2 (18.96%)",
       y = "Dimension 3 (10.10%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")


fviz_pca_biplot(bp_excb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = bp_excb_pca_data$SUB_CATEGORY,
                palette = c("grey80", "grey60", "grey40", "grey20"),
                label = "var") +
  labs(title = "PCA 1 vs. PCA 2 (Backed Point Variant)",
       x = "Dimension 1 (62.35%)",
       y = "Dimension 2 (18.96%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))

fviz_pca_biplot(bp_excb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = bp_excb_pca_data$SUB_CATEGORY,
                palette = c("grey80", "grey60", "grey40", "grey20"),
                label = "var") +
  labs(title = "PCA 2 vs. PCA 3 (Backed Point Variant)",
       x = "Dimension 2 (18.96%)",
       y = "Dimension 3 (10.10%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))

```

```{r bp excb pca boxplots, echo = FALSE, message = FALSE, fig.width = 10, fig.height = 7, fig.align="center"}

bp_excb_pca_x <- cbind(bp_excb_pca_data, bp_excb_pca$ind$coord) %>% 
  select(CODE, CONTEXT, CLASSIFICATION, SUB_CATEGORY, Dim.1, Dim.2, Dim.3, Dim.4, Dim.5) %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5)

bp_excb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = CLASSIFICATION)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

bp_excb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = CODE)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

bp_excb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = SUB_CATEGORY)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

```

# Analysis: Backed Points (inc. burinated pieces)

## Visual and Descriptive Summaries of Data

```{r bp incb clean and display, echo = FALSE, message = FALSE}

bp_data_incb_clean <- bp_data %>%
  mutate(SUB_CATEGORY = fct_recode(SUB_CATEGORY, "Creswellian Point" = "Creswellian/Penknife Point"),
         SUB_CATEGORY = fct_recode(SUB_CATEGORY, "Creswellian Point" = "Penknife Point"),
         SUB_CATEGORY = fct_recode(SUB_CATEGORY, "Creswellian Point" = "Kent Point")) %>%
  rename(BASE_LOCATION = TANG_ORIENTATION)

bp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 46. Descriptive Statistics: Backed Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 47. Descriptive Statistics: Backed Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 48. Descriptive Statistics: Backed Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
bp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 49. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 50. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 51. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 52. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 53. Descriptive Statistics: Backed Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 54. Descriptive Statistics: Backed Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 55. Descriptive Statistics: Backed Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
bp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 56. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 57. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 58. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 59. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 60. Descriptive Statistics: Backed Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 61. Descriptive Statistics: Backed Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 62. Descriptive Statistics: Backed Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
bp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 63. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 64. Descriptive Statistics: Backed Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 65. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

bp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 66. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

```

```{r bp incb summary 1, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}

ggplot(bp_data_incb_clean, aes(CODE, WEIGHT, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Weight (g)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(CODE, LENGTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(CODE, WIDTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Width (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(CODE, ELONGATION, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Elongation (L/W)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(CODE, THICKNESS, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Thickness (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(CODE, TCSA, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSA)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(CODE, TCSP, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSP)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(SUB_CATEGORY, WEIGHT)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Weight (g)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(SUB_CATEGORY, LENGTH)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(SUB_CATEGORY, WIDTH)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Width (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(SUB_CATEGORY, ELONGATION)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Elongation (L/W)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(SUB_CATEGORY, THICKNESS)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Thickness (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(SUB_CATEGORY, TCSA)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSA)", fill = "Taxonomic Unit") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(SUB_CATEGORY, TCSP)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSP)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")
```

```{r bp incb tcsa tcsp plot, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 6, fig.align = "center"}

ggplot(bp_data_incb_clean, aes(TCSA, TCSP, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") +
  stat_ellipse(type = "norm", linetype = 2) +
  lims(x = c(0,100), y = c(0,50)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(TCSA, TCSP, colour = CODE, shape = CODE)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") +
  lims(x = c(0,100), y = c(0,50)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20")) +
  scale_shape_manual(values = c(1,2,3,4,5,6)) +
  theme(legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(TCSA, TCSP, colour = CODE, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") + 
  lims(x = c(0,100), y = c(0,50)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20")) +
  scale_shape_manual(values = c(1,2,3,4,5))

ggplot(bp_data_incb_clean, aes(TCSA, TCSP, colour = CLASSIFICATION, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") + 
  lims(x = c(0,100), y = c(0,50)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  scale_shape_manual(values = c(1,2,3,4,5))

```

```{r bp incb l w plot, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 6, fig.align = "center"}

ggplot(bp_data_incb_clean, aes(WIDTH, LENGTH, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
         geom_point() +
         labs(x = "Backed Point Width (mm)", y = "Backed Point Length (mm)") +
  stat_ellipse(type = "norm", linetype = 2) +
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme(legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(WIDTH, LENGTH, colour = CODE, shape = CODE)) +
         geom_point() +
         labs(x = "Backed Point Width (mm)", y = "Backed Point Length (mm)") +
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20")) +
  scale_shape_manual(values = c(1,2,3,4,5,6)) +
  theme(legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(WIDTH, LENGTH, colour = CODE, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "Backed Point Width (mm)", y = "Backed Point Length (mm)") + 
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20")) +
  scale_shape_manual(values = c(1,2,3,4,5))

ggplot(bp_data_incb_clean, aes(WIDTH, LENGTH, colour = CLASSIFICATION, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "Backed Point Width (mm)", y = "Backed Point Length (mm)") + 
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  scale_shape_manual(values = c(1,2,3,4))

```

```{r bp incb tests, echo = FALSE, message = FALSE, warning = FALSE}

pairwise.wilcox.test(bp_data_incb_clean$WEIGHT, bp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$LENGTH, bp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$WIDTH, bp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$ELONGATION, bp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$THICKNESS, bp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$TCSA, bp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$TCSP, bp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)

pairwise.wilcox.test(bp_data_incb_clean$WEIGHT, bp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$LENGTH, bp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$WIDTH, bp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$ELONGATION, bp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$THICKNESS, bp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$TCSA, bp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$TCSP, bp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)

pairwise.wilcox.test(bp_data_incb_clean$WEIGHT, bp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$LENGTH, bp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$WIDTH, bp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$ELONGATION, bp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$THICKNESS, bp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$TCSA, bp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(bp_data_incb_clean$TCSP, bp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)

```

```{r bp incb summary 2, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}

ggplot(bp_data_incb_clean, aes(fill = BASE_LOCATION, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Base Location", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(fill = BASE_LOCATION, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Base Location", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(bp_data_incb_clean, aes(fill = BASE_LOCATION, x = SUB_CATEGORY)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Base Location", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

```

## Principal Component Analysis (PCA)

```{r bp incb pca, echo = FALSE, message = FALSE}

bp_incb_pca_data <- bp_data_incb_clean %>%
  select(CODE, CONTEXT, CLASSIFICATION, SUB_CATEGORY, WEIGHT, LENGTH, WIDTH, ELONGATION, THICKNESS, TCSA, TCSP)

bp_incb_pca <- PCA(bp_incb_pca_data, graph = FALSE, quali.sup = 1:4)

```

```{r bp incb pca table, echo = FALSE, message = FALSE, warning = FALSE}

as.tibble(get_eig(bp_incb_pca), rownames = "rowname") %>%
  rename(variance = variance.percent,
         cumulative = cumulative.variance.percent)

```

```{r bp incb pca scree, echo = FALSE, message = FALSE, fig.width = 4, fig.height = 3, fig.align="center"}

fviz_screeplot(bp_incb_pca, ylim = c(0,50), barfill = "grey50", barcolor = "grey50", geom = "bar") +
  labs(title = "", x = "Dimension") + 
  theme_minimal()

```

```{r bp incb pca contributions, echo = FALSE, message = FALSE, fig.align="center"}

bp_incb_pca_var <- get_pca_var(bp_incb_pca)

as_tibble(bp_incb_pca_var$contrib, rownames = "rowname") %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5) %>%
  gather("Contribution", "n", 2:6) %>%
  ggplot(aes(n, rowname, fill = rowname)) +
  geom_col(show.legend = FALSE, width = 0.75) +
  labs(title = "Contribution of Variables", x = "Contribution (%)", y = "Variable") +
  scale_fill_grey() +
  facet_wrap(~Contribution, nrow = 1) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))
```

```{r bp incb pca graphs, echo = FALSE, message = FALSE, fig.width = 7, fig.height = 7, fig.align="center"}

fviz_pca_biplot(bp_incb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = bp_incb_pca_data$CLASSIFICATION,
                palette = c("#466995", "grey40", "grey80", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Classification)",
       x = "Dimension 1 (62.99%)",
       y = "Dimension 2 (15.56%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(bp_incb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = bp_incb_pca_data$CLASSIFICATION,
                palette = c("#466995", "grey40", "grey80", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Classification)",
       x = "Dimension 2 (15.56%)",
       y = "Dimension 3 (11.42%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(bp_incb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = bp_incb_pca_data$CODE,
                palette = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Code)",
       x = "Dimension 1 (62.99%)",
       y = "Dimension 2 (15.56%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(bp_incb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = bp_incb_pca_data$CODE,
                palette = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Code)",
       x = "Dimension 2 (15.56%)",
       y = "Dimension 3 (11.42%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")


fviz_pca_biplot(bp_incb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = bp_incb_pca_data$SUB_CATEGORY,
                palette = c("grey90", "grey70", "grey50", "grey30", "grey10"),
                label = "var") +
  labs(title = "PCA 1 vs. PCA 2 (Backed Point Variant)",
       x = "Dimension 1 (62.99%)",
       y = "Dimension 2 (15.56%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))

fviz_pca_biplot(bp_incb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = bp_incb_pca_data$SUB_CATEGORY,
                palette = c("grey90", "grey70", "grey50", "grey30", "grey10"),
                label = "var") +
  labs(title = "PCA 2 vs. PCA 3 (Backed Point Variant)",
       x = "Dimension 2 (15.56%)",
       y = "Dimension 3 (11.42%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))

```

```{r bp incb pca boxplots, echo = FALSE, message = FALSE, fig.width = 10, fig.height = 7, fig.align="center"}

bp_incb_pca_x <- cbind(bp_incb_pca_data, bp_incb_pca$ind$coord) %>% 
  select(CODE, CONTEXT, CLASSIFICATION, SUB_CATEGORY, Dim.1, Dim.2, Dim.3, Dim.4, Dim.5) %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5)

bp_incb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = CLASSIFICATION)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "grey40", "grey80", "#56A3A6")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

bp_incb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = CODE)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("#466995", "grey80", "#56A3A6", "grey60", "grey40", "grey20")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

bp_incb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = SUB_CATEGORY)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

```

# Analysis: Tanged Points (exc. burinated pieces)

## Visual and Descriptive Summaries of Data

```{r tp excb clean and display, echo = FALSE, message = FALSE}

tp_data_excb_clean <- tp_data %>%
  filter(!BURINATION == "Y")

tp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 66. Descriptive Statistics: Tanged Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 67. Descriptive Statistics: Tanged Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 68. Descriptive Statistics: Tanged Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
tp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 69. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 70. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 71. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 72. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 73. Descriptive Statistics: Tanged Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 74. Descriptive Statistics: Tanged Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 75. Descriptive Statistics: Tanged Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
tp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 76. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 77. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 78. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 79. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 80. Descriptive Statistics: Tanged Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 81. Descriptive Statistics: Tanged Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 82. Descriptive Statistics: Tanged Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
tp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 83. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 84. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 85. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 86. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

```

```{r tp excb summary 1, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}

ggplot(tp_data_excb_clean, aes(CODE, WEIGHT, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Weight (g)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(CODE, LENGTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(CODE, WIDTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Width (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(CODE, ELONGATION, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Elongation (L/W)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(CODE, THICKNESS, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Thickness (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(CODE, TCSA, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSA)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(CODE, TCSP, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSP)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(SUB_CATEGORY, WEIGHT)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Weight (g)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(SUB_CATEGORY, LENGTH)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(SUB_CATEGORY, WIDTH)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Width (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(SUB_CATEGORY, ELONGATION)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Elongation (L/W)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(SUB_CATEGORY, THICKNESS)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Thickness (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(SUB_CATEGORY, TCSA)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSA)", fill = "Taxonomic Unit") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(SUB_CATEGORY, TCSP)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSP)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")
```

```{r tp excb tcsa tcsp plot, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 6, fig.align = "center"}

ggplot(tp_data_excb_clean, aes(TCSA, TCSP, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") +
  stat_ellipse(type = "norm", linetype = 2) +
  lims(x = c(0,75), y = c(0,40)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey40", "#56A3A6")) +
  theme(legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(TCSA, TCSP, colour = CODE, shape = CODE)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") +
  lims(x = c(0,75), y = c(0,40)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey90", "#56A3A6", "grey50", "grey30", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5)) +
  theme(legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(TCSA, TCSP, colour = CODE, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") + 
  lims(x = c(0,75), y = c(0,40)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey90", "#56A3A6", "grey50", "grey30", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5))

ggplot(tp_data_excb_clean, aes(TCSA, TCSP, colour = CLASSIFICATION, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") + 
  lims(x = c(0,75), y = c(0,40)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey40", "#56A3A6")) +
  scale_shape_manual(values = c(1,2,3,4,5))

```

```{r tp excb l w plot, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 6, fig.align = "center"}

ggplot(tp_data_excb_clean, aes(WIDTH, LENGTH, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
         geom_point() +
         labs(x = "Tanged Point Width (mm)", y = "Tanged Point Length (mm)") +
  stat_ellipse(type = "norm", linetype = 2) +
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey40", "#56A3A6")) +
  theme(legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(WIDTH, LENGTH, colour = CODE, shape = CODE)) +
         geom_point() +
         labs(x = "Tanged Point Width (mm)", y = "Tanged Point Length (mm)") +
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey90", "#56A3A6", "grey50", "grey30", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5)) +
  theme(legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(WIDTH, LENGTH, colour = CODE, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "Tanged Point Width (mm)", y = "Tanged Point Length (mm)") + 
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey90", "#56A3A6", "grey50", "grey30", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5))

ggplot(tp_data_excb_clean, aes(WIDTH, LENGTH, colour = CLASSIFICATION, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "Tanged Point Width (mm)", y = "Tanged Point Length (mm)") + 
  lims(x = c(0,40), y = c(0,80)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey40", "#56A3A6")) +
  scale_shape_manual(values = c(1,2,3,4,5))

```

```{r tp excb tests, echo = FALSE, message = FALSE, warning = FALSE}

pairwise.wilcox.test(tp_data_excb_clean$WEIGHT, tp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$LENGTH, tp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$WIDTH, tp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$ELONGATION, tp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$THICKNESS, tp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$TCSA, tp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$TCSP, tp_data_excb_clean$CODE, p.adj = "bonf", exact = FALSE)

pairwise.wilcox.test(tp_data_excb_clean$WEIGHT, tp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$LENGTH, tp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$WIDTH, tp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$ELONGATION, tp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$THICKNESS, tp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$TCSA, tp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$TCSP, tp_data_excb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)

pairwise.wilcox.test(tp_data_excb_clean$WEIGHT, tp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$LENGTH, tp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$WIDTH, tp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$ELONGATION, tp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$THICKNESS, tp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$TCSA, tp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_excb_clean$TCSP, tp_data_excb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)

```

```{r tp excb summary 2, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}

ggplot(tp_data_excb_clean, aes(fill = TANG_ORIENTATION, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Tang Orientation", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(fill = TANG_ORIENTATION, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Tang Orientation", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(tp_data_excb_clean, aes(fill = TANG_ORIENTATION, x = SUB_CATEGORY)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Tang Orientation", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

```

## Principal Component Analysis (PCA)

```{r tp excb pca, echo = FALSE, message = FALSE}

tp_excb_pca_data <- tp_data_excb_clean %>%
  select(CODE, CONTEXT, CLASSIFICATION, SUB_CATEGORY, WEIGHT, LENGTH, WIDTH, ELONGATION, THICKNESS, TCSA, TCSP)

tp_excb_pca <- PCA(tp_excb_pca_data, graph = FALSE, quali.sup = 1:4)

```

```{r tp excb pca table, echo = FALSE, message = FALSE, warning = FALSE}

as.tibble(get_eig(tp_excb_pca), rownames = "rowname") %>%
  rename(variance = variance.percent,
         cumulative = cumulative.variance.percent)

```

```{r tp excb pca scree, echo = FALSE, message = FALSE, fig.width = 4, fig.height = 3, fig.align="center"}

fviz_screeplot(tp_excb_pca, ylim = c(0,50), barfill = "grey50", barcolor = "grey50", geom = "bar") +
  labs(title = "", x = "Dimension") + 
  theme_minimal()

```

```{r tp excb pca contributions, echo = FALSE, message = FALSE, fig.align="center"}

tp_excb_pca_var <- get_pca_var(tp_excb_pca)

as_tibble(tp_excb_pca_var$contrib, rownames = "rowname") %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5) %>%
  gather("Contribution", "n", 2:6) %>%
  ggplot(aes(n, rowname, fill = rowname)) +
  geom_col(show.legend = FALSE, width = 0.75) +
  labs(title = "Contribution of Variables", x = "Contribution (%)", y = "Variable") +
  scale_fill_grey() +
  facet_wrap(~Contribution, nrow = 1) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))
```

```{r tp excb pca graphs, echo = FALSE, message = FALSE, fig.width = 7, fig.height = 7, fig.align="center"}

fviz_pca_biplot(tp_excb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = tp_excb_pca_data$CLASSIFICATION,
                palette = c("grey40", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Classification)",
       x = "Dimension 1 (61.07%)",
       y = "Dimension 2 (23.62%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(tp_excb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = tp_excb_pca_data$CLASSIFICATION,
                palette = c("grey40", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Classification)",
       x = "Dimension 2 (23.62%)",
       y = "Dimension 3 (7.57%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(tp_excb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = tp_excb_pca_data$CODE,
                palette = c("grey90", "#56A3A6", "grey50", "grey30", "grey10"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Code)",
       x = "Dimension 1 (61.07%)",
       y = "Dimension 2 (23.62%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(tp_excb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = tp_excb_pca_data$CODE,
                palette = c("grey90", "#56A3A6", "grey50", "grey30", "grey10"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Code)",
       x = "Dimension 2 (23.62%)",
       y = "Dimension 3 (7.57%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")


fviz_pca_biplot(tp_excb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = tp_excb_pca_data$SUB_CATEGORY,
                palette = c("grey90", "grey70", "grey50", "grey30", "grey10"),
                label = "var") +
  labs(title = "PCA 1 vs. PCA 2 (Tanged Point Variant)",
       x = "Dimension 1 (61.07%)",
       y = "Dimension 2 (23.62%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))

fviz_pca_biplot(tp_excb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = tp_excb_pca_data$SUB_CATEGORY,
                palette = c("grey90", "grey70", "grey50", "grey30", "grey10"),
                label = "var") +
  labs(title = "PCA 2 vs. PCA 3 (Tanged Point Variant)",
       x = "Dimension 2 (23.62%)",
       y = "Dimension 3 (7.57%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))

```

```{r tp excb pca boxplots, echo = FALSE, message = FALSE, fig.width = 10, fig.height = 7, fig.align="center"}

tp_excb_pca_x <- cbind(tp_excb_pca_data, tp_excb_pca$ind$coord) %>% 
  select(CODE, CONTEXT, CLASSIFICATION, SUB_CATEGORY, Dim.1, Dim.2, Dim.3, Dim.4, Dim.5) %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5)

tp_excb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = CLASSIFICATION)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

tp_excb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = CODE)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("grey90", "#56A3A6", "grey50", "grey30", "grey10")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

tp_excb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = SUB_CATEGORY)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

```

# Analysis: Tanged Points (inc. burinated pieces)

## Visual and Descriptive Summaries of Data

```{r tp incb clean and display, echo = FALSE, message = FALSE}

tp_data_incb_clean <- tp_data 

tp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 87. Descriptive Statistics: Tanged Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 88. Descriptive Statistics: Tanged Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 89. Descriptive Statistics: Tanged Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
tp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 90. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 91. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 92. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CONTEXT) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 93. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 94. Descriptive Statistics: Tanged Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 95. Descriptive Statistics: Tanged Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 96. Descriptive Statistics: Tanged Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
tp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 97. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 98. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 99. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(CLASSIFICATION) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 100. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(WEIGHT),2),
            max = round(max(WEIGHT),2),
            mean = round(mean(WEIGHT),2),
            med = round(median(WEIGHT),2),
            sd = round(sd(WEIGHT),2),
            var = round(var(WEIGHT),2)) %>% 
  gt() %>%
  tab_header(title = "Table 101. Descriptive Statistics: Tanged Point Weight (g)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_excb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(LENGTH),2),
            max = round(max(LENGTH),2),
            mean = round(mean(LENGTH),2),
            med = round(median(LENGTH,2)),
            sd = round(sd(LENGTH),2),
            var = round(var(LENGTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 81. Descriptive Statistics: Tanged Point Length (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(WIDTH),2),
            max = round(max(WIDTH),2),
            mean = round(mean(WIDTH),2),
            med = round(median(WIDTH),2),
            sd = round(sd(WIDTH),2),
            var = round(var(WIDTH),2)) %>% 
  gt() %>%
  tab_header(title = "Table 102. Descriptive Statistics: Tanged Point Width (mm)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")
            
tp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(ELONGATION),2),
            max = round(max(ELONGATION),2),
            mean = round(mean(ELONGATION),2),
            med = round(median(ELONGATION),2),
            sd = round(sd(ELONGATION),2),
            var = round(var(ELONGATION),2)) %>% 
  gt() %>%
  tab_header(title = "Table 103. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(THICKNESS),2),
            max = round(max(THICKNESS),2),
            mean = round(mean(THICKNESS),2),
            med = round(median(THICKNESS),2),
            sd = round(sd(THICKNESS),2),
            var = round(var(THICKNESS),2)) %>% 
  gt() %>%
  tab_header(title = "Table 104. Descriptive Statistics: Tanged Point Elongation Index (L/W)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSA),2),
            max = round(max(TCSA),2),
            mean = round(mean(TCSA),2),
            med = round(median(TCSA),2),
            sd = round(sd(TCSA),2),
            var = round(var(TCSA),2)) %>% 
  gt() %>%
  tab_header(title = "Table 105. Descriptive Statistics: Tip Cross-Sectional Area (TCSA)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

tp_data_incb_clean %>% 
  group_by(SUB_CATEGORY) %>% 
  summarise(n_obs = n(),
            min = round(min(TCSP),2),
            max = round(max(TCSP),2),
            mean = round(mean(TCSP),2),
            med = round(median(TCSP),2),
            sd = round(sd(TCSP),2),
            var = round(var(TCSP),2)) %>% 
  gt() %>%
  tab_header(title = "Table 106. Descriptive Statistics: Tip Cross-Sectional Perimeter (TCSP)", subtitle = md("&nbsp;")) %>%
  opt_align_table_header(align = "left")

```

```{r tp incb summary 1, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}

ggplot(tp_data_incb_clean, aes(CODE, WEIGHT, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Weight (g)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(CODE, LENGTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(CODE, WIDTH, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Width (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(CODE, ELONGATION, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Elongation (L/W)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(CODE, THICKNESS, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Thickness (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(CODE, TCSA, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSA)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("grey40", "#56A3A6")) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(CODE, TCSP, fill = CLASSIFICATION)) +
         geom_violin() +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSP)", fill = "Taxonomic Unit") +
  theme_minimal() +
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(SUB_CATEGORY, WEIGHT)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Weight (g)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(SUB_CATEGORY, LENGTH)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Length (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(SUB_CATEGORY, WIDTH)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Width (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(SUB_CATEGORY, ELONGATION)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Elongation (L/W)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(SUB_CATEGORY, THICKNESS)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Thickness (mm)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(SUB_CATEGORY, TCSA)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSA)", fill = "Taxonomic Unit") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(SUB_CATEGORY, TCSP)) +
         geom_boxplot(width = 0.3, fill = "grey80") +
         labs(x = "", y = "Tip Cross-Sectional Area (TCSP)", fill = "Taxonomic Unit") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
                                   legend.position = "bottom")
```

```{r tp incb l w plot, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 6, fig.align = "center"}

ggplot(tp_data_incb_clean, aes(WIDTH, LENGTH, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
         geom_point() +
         labs(x = "Tanged Point Width (mm)", y = "Tanged Point Length (mm)") +
  stat_ellipse(type = "norm", linetype = 2) +
  lims(x = c(0,45), y = c(0,90)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey40", "#56A3A6")) + 
  theme(legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(WIDTH, LENGTH, colour = CODE, shape = CODE)) +
         geom_point() +
         labs(x = "Tanged Point Width (mm)", y = "Tanged Point Length (mm)") +
  lims(x = c(0,45), y = c(0,90)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey90", "#56A3A6", "grey50", "grey30", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5)) +
  theme(legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(WIDTH, LENGTH, colour = CODE, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "Tanged Point Width (mm)", y = "Tanged Point Length (mm)") + 
  lims(x = c(0,45), y = c(0,90)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey90", "#56A3A6", "grey50", "grey30", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5))

ggplot(tp_data_incb_clean, aes(WIDTH, LENGTH, colour = CLASSIFICATION, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "Tanged Point Width (mm)", y = "Tanged Point Length (mm)") + 
  lims(x = c(0,45), y = c(0,90)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey40", "#56A3A6")) + 
  scale_shape_manual(values = c(1,2,3,4,5))

```

```{r tp incb tcsa tcsp plot, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 6, fig.align = "center"}

ggplot(tp_data_incb_clean, aes(TCSA, TCSP, colour = CLASSIFICATION, shape = CLASSIFICATION)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") +
  stat_ellipse(type = "norm", linetype = 2) +
  lims(x = c(0,150), y = c(0,75)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey40", "#56A3A6")) + 
  theme(legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(TCSA, TCSP, colour = CODE, shape = CODE)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") +
  lims(x = c(0,150), y = c(0,75)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey90", "#56A3A6", "grey50", "grey30", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5)) +
  theme(legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(TCSA, TCSP, colour = CODE, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") + 
  lims(x = c(0,150), y = c(0,75)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey90", "#56A3A6", "grey50", "grey30", "grey10")) +
  scale_shape_manual(values = c(1,2,3,4,5))

ggplot(tp_data_incb_clean, aes(TCSA, TCSP, colour = CLASSIFICATION, shape = SUB_CATEGORY)) +
         geom_point() +
         labs(x = "TCSA (mm²)", y = "TCSP (mm)") + 
  lims(x = c(0,150), y = c(0,75)) +
  theme_minimal() + 
  scale_colour_manual(values = c("grey40", "#56A3A6")) + 
  scale_shape_manual(values = c(1,2,3,4,5))

```

```{r tp incb tests, echo = FALSE, message = FALSE, warning = FALSE}

pairwise.wilcox.test(tp_data_incb_clean$WEIGHT, tp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$LENGTH, tp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$WIDTH, tp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$ELONGATION, tp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$THICKNESS, tp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$TCSA, tp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$TCSP, tp_data_incb_clean$CODE, p.adj = "bonf", exact = FALSE)

pairwise.wilcox.test(tp_data_incb_clean$WEIGHT, tp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$LENGTH, tp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$WIDTH, tp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$ELONGATION, tp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$THICKNESS, tp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$TCSA, tp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$TCSP, tp_data_incb_clean$CLASSIFICATION, p.adj = "bonf", exact = FALSE)

pairwise.wilcox.test(tp_data_incb_clean$WEIGHT, tp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$LENGTH, tp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$WIDTH, tp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$ELONGATION, tp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$THICKNESS, tp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$TCSA, tp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)
pairwise.wilcox.test(tp_data_incb_clean$TCSP, tp_data_incb_clean$SUB_CATEGORY, p.adj = "bonf", exact = FALSE)

```

```{r tp incb summary 2, echo = FALSE, message = FALSE, fig.width = 6, fig.height = 4, fig.align = "center"}

ggplot(tp_data_incb_clean, aes(fill = TANG_ORIENTATION, x = CODE)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Tang Orientation", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(fill = TANG_ORIENTATION, x = CLASSIFICATION)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Tang Orientation", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

ggplot(tp_data_incb_clean, aes(fill = TANG_ORIENTATION, x = SUB_CATEGORY)) +
  geom_bar(position = position_dodge(preserve = "single"), stat="count") +
  labs(title = "Tang Orientation", x = "", y = "Count", fill = "Type") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

```

## Principal Component Analysis (PCA)

```{r tp incb pca, echo = FALSE, message = FALSE}

tp_incb_pca_data <- tp_data_incb_clean %>%
  select(CODE, CONTEXT, CLASSIFICATION, SUB_CATEGORY, WEIGHT, LENGTH, WIDTH, ELONGATION, THICKNESS, TCSA, TCSP)

tp_incb_pca <- PCA(tp_incb_pca_data, graph = FALSE, quali.sup = 1:4)

```

```{r tp incb pca table, echo = FALSE, message = FALSE, warning = FALSE}

as.tibble(get_eig(tp_incb_pca), rownames = "rowname") %>%
  rename(variance = variance.percent,
         cumulative = cumulative.variance.percent)

```

```{r tp incb pca scree, echo = FALSE, message = FALSE, fig.width = 4, fig.height = 3, fig.align="center"}

fviz_screeplot(tp_incb_pca, ylim = c(0,50), barfill = "grey50", barcolor = "grey50", geom = "bar") +
  labs(title = "", x = "Dimension") + 
  theme_minimal()

```

```{r tp incb pca contributions, echo = FALSE, message = FALSE, fig.align="center"}

tp_incb_pca_var <- get_pca_var(tp_incb_pca)

as_tibble(tp_incb_pca_var$contrib, rownames = "rowname") %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5) %>%
  gather("Contribution", "n", 2:6) %>%
  ggplot(aes(n, rowname, fill = rowname)) +
  geom_col(show.legend = FALSE, width = 0.75) +
  labs(title = "Contribution of Variables", x = "Contribution (%)", y = "Variable") +
  scale_fill_grey() +
  facet_wrap(~Contribution, nrow = 1) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))
```

```{r tp incb pca graphs, echo = FALSE, message = FALSE, fig.width = 7, fig.height = 7, fig.align="center"}

fviz_pca_biplot(tp_incb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = tp_incb_pca_data$CLASSIFICATION,
                palette = c("grey40", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Classification)",
       x = "Dimension 1 (66.79%)",
       y = "Dimension 2 (19.42%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(tp_incb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = tp_incb_pca_data$CLASSIFICATION,
                palette = c("grey40", "#56A3A6"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Classification)",
       x = "Dimension 2 (19.42%)",
       y = "Dimension 3 (9.50%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(tp_incb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = tp_incb_pca_data$CODE,
                palette = c("grey90", "#56A3A6", "grey50", "grey30", "grey10"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 1 vs. PCA 2 (Code)",
       x = "Dimension 1 (66.79%)",
       y = "Dimension 2 (19.42%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")

fviz_pca_biplot(tp_incb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = tp_incb_pca_data$CODE,
                palette = c("grey90", "#56A3A6", "grey50", "grey30", "grey10"),
                label = "var",
                addEllipses = TRUE) +
  labs(title = "PCA 2 vs. PCA 3 (Code)",
       x = "Dimension 2 (19.42%)",
       y = "Dimension 3 (9.50%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom")


fviz_pca_biplot(tp_incb_pca, 
                axes = c(1,2),
                repel = FALSE,
                habillage = tp_incb_pca_data$SUB_CATEGORY,
                palette = c("grey90", "grey70", "grey50", "grey30", "grey10"),
                label = "var") +
  labs(title = "PCA 1 vs. PCA 2 (Tanged Point Variant)",
       x = "Dimension 1 (66.79%)",
       y = "Dimension 2 (19.42%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))

fviz_pca_biplot(tp_incb_pca, 
                axes = c(2,3),
                repel = FALSE,
                habillage = tp_incb_pca_data$SUB_CATEGORY,
                palette = c("grey90", "grey70", "grey50", "grey30", "grey10"),
                label = "var") +
  labs(title = "PCA 2 vs. PCA 3 (Tanged Point Variant)",
       x = "Dimension 2 (19.42%)",
       y = "Dimension 3 (9.50%)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18))

```

```{r tp incb pca boxplots, echo = FALSE, message = FALSE, fig.width = 10, fig.height = 7, fig.align="center"}

tp_incb_pca_x <- cbind(tp_incb_pca_data, tp_incb_pca$ind$coord) %>% 
  select(CODE, CONTEXT, CLASSIFICATION, SUB_CATEGORY, Dim.1, Dim.2, Dim.3, Dim.4, Dim.5) %>%
    rename(`Dimension 1` = Dim.1,
           `Dimension 2` = Dim.2,
           `Dimension 3` = Dim.3,
           `Dimension 4` = Dim.4,
           `Dimension 5` = Dim.5)

tp_incb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = CLASSIFICATION)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("grey40", "#56A3A6")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

tp_incb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = CODE)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_manual(values = c("grey90", "#56A3A6", "grey50", "grey30", "grey10")) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

tp_incb_pca_x %>%
  gather("Value", "n", 5:9) %>%
  ggplot(aes(Value, n, fill = SUB_CATEGORY)) +
  geom_violin() +
  labs(title = "PCA 1:5 (Classification)",
       x = "",
       y = "Value",
       fill = "Classification") +
  scale_fill_grey(start = 0.3, end = 0.9) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 18),
        legend.position = "bottom") 

```
